{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2020-12-01 11:04:10.708 INFO    faiss: Loading faiss with AVX2 support.\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "import spike_queries\n",
    "\n",
    "import pandas as pd\n",
    "import tqdm\n",
    "import pickle\n",
    "import random\n",
    "import itertools\n",
    "\n",
    "import torch\n",
    "from transformers import BertTokenizer\n",
    "from torch.utils.data import TensorDataset, DataLoader, RandomSampler, SequentialSampler\n",
    "from transformers import AutoTokenizer, AutoModel, AutoConfig\n",
    "from transformers import BertForSequenceClassification, AdamW, BertConfig, BertModel, AutoTokenizer, AutoModel, PreTrainedTokenizerFast\n",
    "import numpy as np\n",
    "from typing import List\n",
    "from torch import nn\n",
    "import pytorch_lightning as pl\n",
    "from pytorch_lightning import Trainer\n",
    "from torch.utils.data import DataLoader\n",
    "from torch.utils.data import Dataset\n",
    "from typing import Dict, Tuple\n",
    "from scipy.spatial.distance import cosine as cosine_distance\n",
    "from collections import defaultdict\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"queries2.txt\", \"r\") as f:\n",
    "    queries = f.readlines()\n",
    "    queries = [l.strip() for l in queries]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i,q in enumerate(queries):\n",
    "    try:\n",
    "        assert len(q.split(\"\\t\")) == 2\n",
    "    except:\n",
    "        print(i, q, len(q.split(\"\\t\")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i,q in enumerate(queries):\n",
    "    a = q.count(\"arg1:\") == q.count(\"arg2:\") == 1\n",
    "    if not a:\n",
    "        print(i,q)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<E|U+C>arg1:virus $infection $causes a <E|U+C>arg2:condition\t1\n",
      "<E|U+C>arg1:virus $infection, which $causes a <E|U+C>arg2:condition\t1\n",
      "<E|U+C>arg1:virus $infection $results $in <E|U+C>arg2:condition\t1\n",
      "<E|U+C>arg1:virus $infection, which $results $in <E|U+C>arg2:condition\t1\n",
      "<E|U+C>arg2:condition $is $the $result of <E|U+C>arg1:virus $infection\t1\n",
      "<E|U+C>arg2:condition, which $is $the $result of <E|U+C>arg1:virus $infection\t1\n",
      "<E|U+C>arg2:condition $is $[l=cause|trigger]caused $by <E|U+C>arg1:virus $infection\t1\n",
      "as a result of <E|U+C>arg1:virus $infection, <E|U+C>arg2:condition $can $develop\t1\n",
      "infected <E|U+C>arg1:patients $usually $develop <E|U+C>arg2:[entity]respiratory illness\t2\n",
      "infected <E|U+C>arg1:patients $show $signs $of <E|U+C>arg2:[entity]respiratory illness\t2\n",
      "<E|U+C>arg2:[entity]respiratory illness $is $observed in infected <E|U+C>arg1:patients\t2\n",
      "a <E|U+C>arg1:subset of $patients $progress to <E|U+C>arg2:hemorrhagic fever\t3\n",
      "a <E|U+C>arg1:subset $of $patients $show $signs $of <E|U+C>arg2:hemorrhagic fever\t3\n",
      "people with <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma are $susceptible to <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t4\n",
      "<E|U+C>arg1:something $[l]are $susceptible $to <E|U+C>arg2:something\t4\n",
      "<E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma :is $more $common in people with <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t4\n",
      "<E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma :is $more $common in <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma patients\t4\n",
      "<E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma :is a common $comorbidity in people with <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t4\n",
      "the $comorbidity of <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma and <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t4\n",
      "<E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma :is a common $comorbidity in <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma patients\t4\n",
      "<E|U+C>arg1:asthmatic $patients are $[lemma=prone|susceptible]susceptible $to <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma disease\t4\n",
      "<E|U+C>arg1:low blood oxygen levels are a $risk factor in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION|COVID-19]COVID-19 infection\t5\n",
      "<E|U+C>arg1:something, a $risk $factor $for <E|U+C>arg2:something\t5\n",
      "<E|U+C>arg1:low blood oxygen levels poses $risk in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION|COVID-19]COVID-19 $infection\t5\n",
      "<E|U+C>arg1:[word]age is another $risk $factor for both morbidity and mortality in <E|U+C>arg2:COVID-19 patients\t5\n",
      "<E|U+C>arg1:[word]obesity is another $risk $factor for both morbidity and mortality in <E|U+C>arg2:COVID-19 patients\t5\n",
      "<E|U+C>arg1:low blood oxygen levels are a $risk factor in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t5\n",
      "<E|U+C>arg1:something is identified as a $risk $factor for <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t5\n",
      "<E|U+C>arg1:low blood oxygen levels poses $risk in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t5\n",
      "<E|U+C>arg1:[word]age is another $risk $factor for both morbidity and mortality in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t5\n",
      "<E|U+C>arg1:[word]Obesity is another $risk $factor for both morbidity and mortality in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma patients\t5\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma is a $risk $factor in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t5\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol is $not useful for treating <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t6\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol is $not a $useful treatment for <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t6\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol is $not $effective in treating <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t6\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol $is $not $effective against <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t6\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol $is $not $recommended against <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t6\n",
      "<E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $should $not $be $treated with <E|U+C>arg1:something\t6\n",
      "<E|U+C>arg1:something $is $contraindicated in $patients with <E|U+C>arg2:[e]asthme\t6\n",
      "<E|U+C>arg1:something $is $contraindicated in <E|U+C>arg2:[e=DISEASE]asthma\t6\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol $should $not $be $used to $treat <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t6\n",
      "<E|U+C>arg1:[e]stroke is a $complication of <E|U+C>arg2:[e=COVID-19|DISEASE|ORGANISM|ORGANISM_SUBSTANCE]COVID-19 infection\t7\n",
      "<E|U+C>arg1:[e]stroke is a $complication of <E|U+C>arg2:flu :vaccination\t7\n",
      "<E|U+C>arg1:[e]stroke, which is a $complication of <E|U+C>arg2:flu :vaccination\t7\n",
      "among the $complications trigerred $by <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma is <E|U+C>arg1:[e]stroke\t7\n",
      "the $complications of <E|U+C>arg2:flu $infection include <E|U+C>arg1:[e]stroke\t7\n",
      "the recommended $quarantine period in <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma infection is <E|U+C>arg2:14 :[w]days\t8\n",
      "patients diagnosed with <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma should be $quarantined for <E|U+C>arg2:14 :[w]days\t8\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma patients should be $quarantined for <E|U+C>arg2:14 :[w]days\t8\n",
      "<E|U+C>arg1:[e=COVID-19|ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]COVID-19 patients should be $quarantined for <E|U+C>arg2:14 :[w]days\t8\n",
      "<E|U+C>arg1:[e=COVID-19|ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]COVID-19 $[l=infect|invade|damage]infects <E|U+C>arg2:lung $cells\t9\n",
      "<E|U+C>arg2:lung $cells are $infected by <E|U+C>arg1:[e=COVID_19|ORGANISM|ORGANISM_SUBSTANCE|ORGANISM_SUBDIVISION|DISEASE]COVID-19\t9\n",
      "<E|U+C>arg2:lung $[l=cell|tissue]cells $infected by <E|U+C>arg1:[e]COVID-19\t9\n",
      "<E|U+C>arg1:influenza $virus $[l]infects <E|U+C>arg2:lung $cells\t9\n",
      "<E|U+C>arg1:influenza $virus $[l]invades to <E|U+C>arg2:lung $cells\t9\n",
      "<E|U+C>arg1:influenza $virus $[l]enters the <E|U+C>arg2:lung $cells\t9\n",
      "<E|U+C>arg1:[e=COVID-19|ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]COVID-19 $[l=invade|enter|infect|damage]invades <E|U+C>arg2:lung $tissue\t9\n",
      "<E|U+C>arg2:lung $tissue is $[l=infect|invade|damage|enter]infected by <E|U+C>arg1:[e=COVID-19|ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]COVID-19\t9\n",
      "<E|U+C>arg1:influenza $virus $[l=invade|enter|infect|damage]invades to <E|U+C>arg2:lung $tissue\t9\n",
      "<E|U+C>arg1:something $[l=activate|bind|regulate|inhibit|damage|enter]activates the <E|U+C>arg2:ATP $receptor\t10\n",
      "<E|U+C>arg1:something $[l]binds $to the <E|U+C>arg2:ATP $receptor\t10\n",
      "<E|U+C>arg1:[e]COVID-19 activates the <E|U+C>arg2:ATP $receptor\t10\n",
      "<E|U+C>arg1:[e=ORGANISM_SUBSTANCE]COVID-19 activates the <E|U+C>arg2:ATP $receptor\t10\n",
      "<E|U+C>arg2:ATP $receptor is activated by <E|U+C>arg1:[e=COVID-19|ORGANISM|ORGANISM_SUBSTANCE]COVID-19\t10\n",
      "<E|U+C>arg2:ATP $receptor, which is activated by <E|U+C>arg1:[e=COVID-19|ORGANISM|ORGANISM_SUBSTANCE]COVID-19\t10\n",
      "<E|U+C>arg2:ATP $receptor is $[l]activated $by <E|U+C>arg1:something\t10\n",
      "<E|U+C>arg2:ATP $receptor, which is $[l]activated $by <E|U+C>arg1:something\t10\n",
      "The <E|U+C>arg1:flu $virus $originated in <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]bats\t11\n",
      "The <E|U+C>arg1:flu $virus, which $originated in <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]bats\t11\n",
      "<E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]bats are the $hosts of the <E|U+C>arg1:flu $virus\t11\n",
      "<E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]bats are a $vector $of $transmission for <E|U+C>arg1:something\t11\n",
      "<E|U+C>arg1:flu $virus $transmitted $from <E|U+C>arg2:[e]bats\t11\n",
      "<E|U+C>arg1:something $transmitted $from <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]something\t11\n",
      "$transmission $of <E|U+C>arg1:something $from <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]bats\t11\n",
      "<E|U+C>arg2:[e]bats are a $reservoir $of <E|U+C>arg1:[e]something\t11\n",
      "<E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]bats $[l]shed <E|U+C>arg1:something $virus\t11\n",
      "<E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]bats $[l]shed <E|U+C>arg1:[e]COVID-19\t11\n",
      "<E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]bats $[l]shed <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBSTANCE|ORGANISM_SUBDIVISION]COVID-19\t11\n",
      "<E|U+C>arg1:flu $virus $[l=evolve|diverge]evolved $from <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]something\t12\n",
      "<E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]flu $[l=evolve|diverge]evolved $from <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]something\t12\n",
      "<E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]flu, that $[l=evolve|diverge]evolved $from <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]something\t12\n",
      "the $[w=evolution|divergence]evolution $of <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBSTANCE]flu $from <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]something\t12\n",
      "The $mechanism $of $action of <E|U+C>arg1:paracetamol is <E|U+C>arg2:activation of something\t13\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol $[lemma=word|act]works $by <E|U+C>arg2:activation of glocuse\t13\n",
      "<E|U+C>arg2:inhibition of kinases is the $[lemma=mode|mechanism]mode $of $action of <E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol\t13\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol $exerts its influnce $by <E|U+C>arg2:ligand activation\t13\n",
      "the $[lemma=effect|influence|action]effect of <E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol is $attributed to <E|U+C>arg2:ligand activation\t13\n",
      "something can be $[l=transmit|cross]transmitted $from <E|U+C>arg1:something $to <E|U+C>arg2:something\t14\n",
      "<E|U+C>arg1:someone can $[l=transmit|cross]transmit something $to <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]someone\t14\n",
      "$[e=COVID-18|ORGANISM|ORGANISM_SUBSTANCE]COVID-19 $[l]crosses from <E|U+C>arg1:someone $to <E|U+C>arg2:someone\t14\n",
      "something, $[l]transmitted $to <E|U+C>arg2:someone $from <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]someone\t14\n",
      "$[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]COVID-19 is $transmitted from <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]someone $to <E|U+C>arg2:someone\t14\n",
      "$[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]COVID-19, that is $transmitted from <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]someone $to <E|U+C>arg2:someone\t14\n",
      "$transmission $of something $from <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBSTANCE]bats $to <E|U+C>arg2:someone\t14\n",
      "<E|U+C>arg1:SCD $[l]genotypes share high $risk for <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg1:[e=DNA|GENE_OR_GENE_PRODUCT]SCD share high $risk for <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg1:SCD $[l]genotypes is a $risk factor for <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg1:[e=DNA|GENE_OR_GENE_PRODUCT]SCD is a $risk $factor for <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg1:SCD $[l]genotypes provide $protection against <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg1:[e=DNA|GENE_OR_GENE_PRODUCT]SCD provide $protection against <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg1:SCD $[l]genotypes provide $immunity against <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg2:something is $more $common in <E|U+C>arg1:SCD $[l]genotypes\t15\n",
      "<E|U+C>arg1:SCD $[l]polymorphism share high $risk for <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg1:SCD $[l]polymorphism provide $protection against <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg1:SCD $[l]polymorphism provide $immunity against <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg2:something is $more $common in <E|U+C>arg1:SCD $[l]polymorphism\t15\n",
      "<E|U+C>arg1:SCD $[l]polymorphism is a $risk factor for <E|U+C>arg2:something\t15\n",
      "<E|U+C>arg1:white $cells $[l]attack the <E|U+C>arg2:flu $virus\t16\n",
      "<E|U+C>arg1:white $cells $[l]prevent the <E|U+C>arg2:flu $infection\t16\n",
      "<E|U+C>arg1:white $cells $[l]provide $protection against <E|U+C>arg2:flu $virus\t16\n",
      "<E|U+C>arg1:white $cells $[l]protect against <E|U+C>arg2:flu $virus\t16\n",
      "<E|U+C>arg1:white $cells $[l]provide $protection against <E|U+C>arg2:flu $infection\t16\n",
      "<E|U+C>arg1:white $cells $[l]protect against <E|U+C>arg2:flu $infection\t16\n",
      "<E|U+C>arg1:white $cells $[l]destroy the <E|U+C>arg2:flu $virus\t 16\n",
      "<E|U+C>arg2:flu $virus is $[l=attack|destroy|incapacitate|neutralize]attacked by <E|U+C>arg1:white cells\t16\n",
      "<E|U+C>arg2:flu $virus is $[l]destroyed by <E|U+C>arg1:white $cells\t16\n",
      "<E|U+C>arg1:something is a $countermeasure against <E|U+C>arg2:something\t17\n",
      "<E|U+C>arg1:something, a $countermeasure against <E|U+C>arg2:something\t17\n",
      "<E|U+C>arg1:something, which is a $countermeasure against <E|U+C>arg2:something\t17\n",
      "<E|U+C>arg1:something $is $useful $against <E|U+C>arg2:something\t17\n",
      "<E|U+C>arg1:something $[l]contributes to the $spread of <E|U+C>arg2:something infection\t18\n",
      "<E|U+C>arg1:something $[l]eases the $spread of <E|U+C>arg2:something infection\t18\n",
      "<E|U+C>arg1:something $[l=prevent|help|facilitate|ease]prevents the $spread of <E|U+C>arg2:something infection\t18\n",
      "<E|U+C>arg1:something, which $[l=prevent|help|facilitate|ease]prevents the $spread of <E|U+C>arg2:something infection\t18\n",
      "<E|U+C>arg1:something will $[l=prevent|help|ease|facilitate]help <E|U+C>arg2:something infection $to $spread\t18\n",
      "<E|U+C>arg1:something $[l=help|prevent|ease|facilitate]helps <E|U+C>arg2:something infection $to $spread\t18\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol $treats <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t19\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol $is $useful $against <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t19\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol, a $treatment for <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t19\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol $is $a $treatment for <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma.\t 19\n",
      "<E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $should $be $treated with <E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol\t19\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol is used $to $treat <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t19\n",
      "<E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $is $treated with <E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol\t19\n",
      "<E|U+C>arg1:something is $genetically $related to <E|U+C>arg2:something\t 20\n",
      "<E|U+C>arg1:something, which is $genetically $related to <E|U+C>arg2:something\t20\n",
      "<E|U+C>arg1:something $and <E|U+C>arg2:something are $genetically $related\t20\n",
      "<E|U+C>arg1:something $and <E|U+C>arg2:something share $a $common $[w=origin|ancestry|ancestor]ancestry\t20\n",
      "the $genetic $relation between <E|U+C>arg1:something $and <E|U+C>arg2:something\t20\n",
      "<E|U+C>arg1:flu $virus interacts with the <E|U+C>arg2:immune $system\t21\n",
      "<E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]flu interacts with the <E|U+C>arg2:immune $system\t21\n",
      "<E|U+C>arg1:flu $[w=virus|infection]infection influences the <E|U+C>arg2:immune $system\t21\n",
      "<E|U+C>arg1:flu $[w=infection|virus]infection, which influences the <E|U+C>arg2:immune $system\t21\n",
      "the <E|U+C>arg2:immune $system is influenced from <E|U+C>arg1:flu $infection\t21\n",
      "the <E|U+C>arg2:immune $system is influenced from <E|U+C>arg1:flu $virus\t21\n",
      "<E|U+C>arg1:glocuse $blood $levels are $[l=reduce|elevate]elevated in <E|U+C>arg2:something\t22\n",
      "<E|U+C>arg1:glocuse $blood $levels are an $indication of <E|U+C>arg2:something\t22\n",
      "$high $blood <E|U+C>arg1:glocuse is a $sign of <E|U+C>arg2:something\t22\n",
      "low $blood <E|U+C>arg1:glocuse is a $[l=sign|indication|symptom]sign of <E|U+C>arg2:something\t22\n",
      "low $blood <E|U+C>arg1:glocuse, which is a $[l=sign|indication|symptom]sign of <E|U+C>arg2:something\t22\n",
      "low $blood <E|U+C>arg1:glocuse, a $[l=sign|indication|symptom]sign of <E|U+C>arg2:something\t22\n",
      "the $symptoms of <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma are <E|U+C>arg2:[e]stroke\t23\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $symptoms $such $as <E|U+C>arg2:[e]stroke\t23\n",
      "<E|U+C>arg2:[e]stroke, $a $[l=symptom|sign]symptom $of <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t23\n",
      "the $[l=sign|symptom]signs of <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma are <E|U+C>arg2:[e]stroke\t23\n",
      "the $[l=symptoms|sign]symptoms of <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $include <E|U+C>arg2:something\t23\n",
      "<E|U+C>arg2:something is $a $[w=symptom|sign]sign $of <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t23\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma is $manifested by <E|U+C>arg2:something\t23\n",
      "people $infected $with <E|U+C>arg1:something $[l]experience <E|U+C>arg2:something\t23\n",
      "people $infected $with <E|U+C>arg1:something $[l]show $signs $of <E|U+C>arg2:something\t23\n",
      "the $clinical $manifestations of <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma include <E|U+C>arg2:something\t23\n",
      "<E|U+C>arg2:something is a $symptom of <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t23\n",
      "<E|U+C>arg1:something $[l=encourage|trigger|inhibit]encourages the $production of <E|U+C>arg2:something\t24\n",
      "<E|U+C>arg1:something, that $[l=encourage|trigger|inhibit]encourages the $production of <E|U+C>arg2:something\t24\n",
      "<E|U+C>arg1:something $[l=trigger|encourage]triggers the body $to $produce <E|U+C>arg2:something\t24\n",
      "<E|U+C>arg1:something can $[l=increase|improve|help]increase $survival in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION|CANCER]something\t25\n",
      "<E|U+C>arg1:something can $[w=decrease|prevent|increase]decrease $death rates in <E|U+C>arg2:[e=DISEASE|CANCER|PATHOLOGICAL_FORMATION]something\t25\n",
      "$survival in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION|CANCER]something is $improved by arg1:something\t25\n",
      "<E|U+C>arg1:something, that improved $survival in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION|CANCER]something\t25\n",
      "$[lemma=treatment|management]treatment for <E|U+C>arg1:something should $focus on <E|U+C>arg2:something\t26\n",
      "<E|U+C>arg1:something is $pivotal in <E|U+C>arg2:[e=DISEASE]something :[lemma=treatment|management]treatment\t26\n",
      "<E|U+C>arg2:something is a $main $part in <E|U+C>arg1:something $treatment\t26\n",
      "<E|U+C>arg1:something $[lemma=cause|induce]causes $death $[w=within|after]within <E|U+C>arg2:[e]10 days\t27\n",
      "animals $infected with <E|U+C>arg1:something $died $within <E|U+C>arg2:something\t27\n",
      "in <E|U+C>arg1:something $infection $death $occurred $after <E|U+C>arg2:something\t27\n",
      "the $duration of <E|U+C>arg1:[e=DISEASE]something is <E|U+C>arg2:[e]10 days\t28\n",
      "<E|U+C>arg1:something $infection $[l]lasts <E|U+C>arg2:10 days\t28\n",
      "animals $infected with <E|U+C>arg1:something $[l]recover after <E|U+C>arg2:10 days\t28\n",
      "<E|U+C>arg1:something $is a $first $symptom $of <E|U+C>arg2:something\t29\n",
      "<E|U+C>arg1:something $is a $first $sign $of <E|U+C>arg2:something\t29\n",
      "the $first $symptom $of <E|U+C>arg2:something $[l]is <E|U+C>arg1:something\t29\n",
      "the $first $sign $of <E|U+C>arg2:something $[l]is <E|U+C>arg1:something\t29\n",
      "<E|U+C>arg2:something $is $initially $manifested with <E|U+C>arg1:something\t29\n",
      "<E|U+C>arg2:[e=DISEASE]something $initially $[l]presents with <E|U+C>arg1:something\t29\n",
      "<E|U+C>arg2:something $[l]is the $initial $manifestation of <E|U+C>arg1:something\t29\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $patients are $contagious $for <E|U+C>arg2:10 days\t30\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $patients $remain $contagious $for <E|U+C>arg2:10 days\t30\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $patients are $infectious $for <E|U+C>arg2:10 days\t30\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $patients $remain $infectious $for <E|U+C>arg2:10 days\t30\n",
      "patients infected with <E|U+C>arg1:[e]COVID-19 are $infectious for <E|U+C>arg2:10 days\t30\n",
      "<E|U+C>arg1:asthma patients are $symptomatic $for <E|U+C>arg2:something days\t31\n",
      "<E|U+C>arg1:[e]COVID-19 $patients remain $symptomatic $for <E|U+C>arg2:something days\t31\n",
      "<E|U+C>arg1:[e]COVID-19 $patients $show $symptoms $for <E|U+C>arg2:something $days\t31\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $patients remain $symptomatic $for <E|U+C>arg2:something $days\t31\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $patients $show $symptoms $for <E|U+C>arg2:something $days\t31\n",
      "the $symptoms of <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $last for <E|U+C>arg2:10 days\t31\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $symptoms $last for <E|U+C>arg2:10 days\t31\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]something $does $not $transmit to <E|U+C>arg2:someone\t32\n",
      "<E|U+C>arg1:something $infection $does $not $transmit to <E|U+C>arg2:someone\t32\n",
      "<E|U+C>arg1:something $virus $does $not $transmit to <E|U+C>arg2:someone\t32\n",
      "<E|U+C>arg1:something $virus $does $not $infect <E|U+C>arg2:someone\t32\n",
      "<E|U+C>arg2:someone $[l]is $not $affected from <E|U+C>arg1:something $infection\t32\n",
      "<E|U+C>arg1:something $virus $does $not $affect <E|U+C>arg2:someone\t32\n",
      "<E|U+C>arg2:someone $[l]are $immune $to <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBSTANCE]something\t32\n",
      "<E|U+C>arg1:something $virus affects the <E|U+C>arg2:something $ligand\t33\n",
      "<E|U+C>arg1:something $virus affects the <E|U+C>arg2:[e]respiratory $system\t33\n",
      "<E|U+C>arg1:something $infection affects the <E|U+C>arg2:something $ligand\t33\n",
      "<E|U+C>arg1:something $[l]infection affects the <E|U+C>arg2:[e]respiratory $system\t33\n",
      "<E|U+C>arg1:something $was first $isolated $from <E|U+C>arg2:soil\t34\n",
      "<E|U+C>arg1:something $was $first $detected $in <E|U+C>arg2:soil\t34\n",
      "<E|U+C>arg1:something $was first $discovered $in <E|U+C>arg2:human :saliva\t34\n",
      "<E|U+C>arg1:something $[l]spreads $[word=through]by <E|U+C>arg2:something\t35\n",
      "<E|U+C>arg2:something is a $medium of $transmission of <E|U+C>arg1:something\t35\n",
      "<E|U+C>arg1:something $virus can spread through <E|U+C>arg2:[w]droplets\t35\n",
      "the $modes $of $transmission of <E|U+C>arg1:something $virus $include <E|U+C>arg2:[w]droplets\t35\n",
      "the $modes $of $transmission of <E|U+C>arg1:something $virus $include <E|U+C>arg2:airborne particles\t35\n",
      "<E|U+C>arg1:[e]COVID-19 can spread through <E|U+C>arg2:[w]droplets\t35\n",
      "<E|U+C>arg1:[e]COVID-19 can spread through <E|U+C>arg2:[w]contact\t35\n",
      "<E|U+C>arg1:[e]COVID-19 can spread through <E|U+C>arg2:[w]airborne particles\t35\n",
      "the primary $[l]route $of <E|U+C>arg1:[e]COVID-19 infection is <E|U+C>arg2:something\t35\n",
      "the primary $[l]route $of $transmission of <E|U+C>arg1:[e]COVID-19 infection is <E|U+C>arg2:something\t35\n",
      "the primary $[l]route $of $infection of <E|U+C>arg1:[e]COVID-19 is <E|U+C>arg2:something\t35\n",
      "the primary $[l]route $of <E|U+C>arg1:something $infection is <E|U+C>arg2:something\t35\n",
      "the $modes $of $transmission of <E|U+C>arg1:[e]COVID-19 $include <E|U+C>arg2:[w]droplets\t35\n",
      "<E|U+C>arg2:[w]droplet $transmission of <E|U+C>arg1:[e]COVID-19\t35\n",
      "<E|U+C>arg2:[w]droplet $transmission of <E|U+C>arg1:flu $virus\t35\n",
      "arg2:[w]airborne $transmission of <E|U+C>arg1:[e]COVID-19\t35\n",
      "arg2:[w]airborne $transmission of <E|U+C>arg1:flu $virus\t35\n",
      "the <E|U+C>arg1:flu $virus is <E|U+C>arg2:[w]airborne\t35\n",
      "the <E|U+C>arg1:[e]COVID-19 is <E|U+C>arg2:[w]airborne\t35\n",
      "<E|U+C>arg1:asthma $is $not $a $risk $factor for <E|U+C>arg2:asthma\t36\n",
      "<E|U+C>arg1:asthma $poses $no $risk for <E|U+C>arg2:asthma\t36\n",
      "<E|U+C>arg1:asthma patients $are $not $suspectible to <E|U+C>arg2:asthma\t36\n",
      "<E|U+C>arg1:[e=COVID-19|DISEASE|ORGANISM]COVID-19 $fatality $rates are <E|U+C>arg2:something\t37\n",
      "<E|U+C>arg2:[e]10 $% of <E|U+C>arg1:[e=COVID-19|DISEASE|ORGANISM]COVID-19 patients $die\t37\n",
      "<E|U+C>arg1:[e=COVID-19|ORGANISM]COVID-19 $death $rates are <E|U+C>arg2:something\t37\n",
      "<E|U+C>arg1:[e=COVID-19|DISEASE|ORGANISM]COVID-19 $results $in $death in <E|U+C>arg2:19 $% of the cases\t37\n",
      "<E|U+C>arg1:[e=COVID-19|DISEASE|ORGANISM]COVID-19 $results $in $death in <E|U+C>arg2:19% of the cases\t37\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $death $rates are <E|U+C>arg2:[e]9 $%\t37\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $fatality $rates are <E|U+C>arg2:[e]9 $%\t37\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $causes $death in <E|U+C>arg2:[e]9 $% of the cases\t37\n",
      "<E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $results $in $death in <E|U+C>arg2:[e]9 $% of the cases\t37\n",
      "the $enzyme <E|U+C>arg1:A $[l]interacts with the $enzyme <E|U+C>arg2:B\t38\n",
      "the $enzyme <E|U+C>arg1:A $[l]activates the $enzyme <E|U+C>arg2:B\t38\n",
      "the $enzyme <E|U+C>arg1:A $[l]regulates the $enzyme <E|U+C>arg2:B\t38\n",
      "the $enzyme <E|U+C>arg1:A $[l]inhibits the $enzyme <E|U+C>arg2:B\t38\n",
      "the $protein <E|U+C>arg1:A $[l]interacts with the $protein <E|U+C>arg2:B\t38\n",
      "the <E|U+C>arg1:[e=PROTEIN]A $[l=interact]interacts with <E|U+C>arg2:[e=PROTEIN]B\t38\n",
      "the <E|U+C>arg1:[e=PROTEIN]A $[l=influence|inhibit|activate|regulate]activate <E|U+C>arg2:[e=PROTEIN]B\t38\n",
      "the $protein <E|U+C>arg1:A $[l=activate|regulate|inhibit]activates the $protein <E|U+C>arg2:B\t38\n",
      "$protein-protein $interaction between <E|U+C>arg1:something :and <E|U+C>arg2:something\t38\n",
      "the $[l]interaction between <E|U+C>arg1:[e=PROTEIN]A and <E|U+C>arg2:[e=PROTEIN]B\t38\n",
      "<E|U+C>arg1:something $[l]inhibits the <E|U+C>arg2:something $enzyme\t39\n",
      "<E|U+C>arg2:something $enzyme is $inhibited by <E|U+C>arg1:something\t39\n",
      "<E|U+C>arg1:something $[l]inhibits the <E|U+C>arg2:something $protein\t39\n",
      "<E|U+C>arg2:something $protein is $inhibited by <E|U+C>arg1:something\t39\n",
      "<E|U+C>arg1:something $is an $inhibitor of <E|U+C>arg2:something\t39\n",
      "$outbreak $of <E|U+C>arg2:[e]MERS-CoV $occurred $in <E|U+C>arg1:[e]France\t40\n",
      "<E|U+C>arg2:[e]MERS-CoV $pandemic occurred $in <E|U+C>arg1:[e]France\t40\n",
      "<E|U+C>arg2:[e]MERS-CoV $epidemic occurred $in <E|U+C>arg1:[e]France\t40\n",
      "$outbreak $of <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $occurred $in <E|U+C>arg1:[e]France\t40\n",
      "<E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $pandemic $[lemma=start|begin|occur|report]began $in <E|U+C>arg1:[e]France\t40\n",
      "<E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $epidemic $[lemma=start|begin|occur|report]began $in <E|U+C>arg1:[e]France\t40\n",
      "$outbreak of <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma $in <E|U+C>arg1:[e]France\t40\n",
      "$outbreak $of <E|U+C>arg2:[e]MERS-CoV $in <E|U+C>arg1:[e]France\t40\n",
      "<E|U+C>arg1:something $[l=induce|trigger|cause|create]induces $[l=immunity|resistence]immunity $against <E|U+C>arg2:something\t41\n",
      "$immunity against <E|U+C>arg1:something can be $[l]achieved $using <E|U+C>arg2:something\t41\n",
      "patients treated with <E|U+C>arg1:something $showed $[l=immunity|resistence]immunity $against <E|U+C>arg2:something\t41\n",
      "<E|U+C>arg1:something $[l]is $a $pretreatment for <E|U+C>arg2:something\t42\n",
      "<E|U+C>arg1:something $[l]is $a $prophylactic treatment for <E|U+C>arg2:something\t42\n",
      "<E|U+C>arg1:something is a $[l]prophylactic $[w=against|for]for <E|U+C>arg2:something\t42\n",
      "<E|U+C>arg1:something is currently $under $research $for <E|U+C>arg2:asthma\t43\n",
      "<E|U+C>arg1:something is an $experimental $[l=treatment|agent|medicine|drug|vaccine]treatment $for <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t43\n",
      "<E|U+C>arg1:something is studied as a $future $treatment $for <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t43\n",
      "<E|U+C>arg1:something is a $future $treatment $for <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t43\n",
      "<E|U+C>arg1:something is $[l=ivestigate|study|evaluate]investigated as a $[l=treatment|agent|medicine|drug|vaccine]treatment $for <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t43\n",
      "<E|U+C>arg1:something is $a $potential $[l=treatment|medicine|drug|vaccine]treatment $for <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\t43\n",
      "<E|U+C>arg1:something can be $[l=detect|identify]identified $[w=with|by]with <E|U+C>arg2:microscope\t44\n",
      "<E|U+C>arg1:something is $[l=detect|identify]identified $[w=with|by]with <E|U+C>arg2:microscope\t44\n",
      "<E|U+C>arg2:HPLC was $[l]performed $to $[w=identify|detect]identify <E|U+C>arg1:something\t44\n",
      "<E|U+C>arg2:microscope was $used to $screen $for <E|U+C>arg1:something\t44\n",
      "<E|U+C>arg1:something was $[w=described|reported|observed]described $by <E|U+C>arg2:someone\t45\n",
      "<E|U+C>arg1:something was first $[w=described|reported|observed]described $by <E|U+C>arg2:someone\t45\n",
      "<E|U+C>arg2:someone $first $[w=described|reported|observed]described <E|U+C>arg1:something\t 45\n",
      "<E|U+C>arg1:something was $named $by <E|U+C>arg2:someone\t45\n",
      "<E|U+C>arg2:someone $discovered <E|U+C>arg1:something\t45\n",
      "<E|U+C>arg1:flu $virus $[l]persists for <E|U+C>arg2:10 $[lemma=day|week|month]days\t46\n",
      "<E|U+C>arg1:[e=DISEASE|ORGANISM]COVID-19 $[l]persists for <E|U+C>arg2:10 $[lemma=day|week|month]days\t46\n",
      "<E|U+C>arg1:[e]COVID-19 $[l]persists for <E|U+C>arg2:10 $[lemma=day|week|month]days\t46\n",
      "<E|U+C>arg1:[e]COVID-19's $persistence is <E|U+C>arg2:10 $[lemma=day|week|month]days\t46\n",
      "a study shows $persistence of <E|U+C>arg1:flu $virus for <E|U+C>arg2:10 $[lemma=day|week|month]days\t46\n",
      "a study shows $persistence of <E|U+C>arg1:[e]COVID-19 for <E|U+C>arg2:10 $[lemma=day|week|month]days\t46\n",
      "a study shows $persistence of <E|U+C>arg1:[e=DISEASE|ORGANISM|ORGANISM|ORGANISM_SUBSTANCE]COVID-19 for <E|U+C>arg2:10 $[lemma=day|week|month]days\t46\n",
      "<E|U+C>arg1:[e]COVID-19 $persists $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:[e]COVID-19 $can $survive $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:[e]COVID-19 $[l]remains $viable $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:[e]COVID-19 $[l]can $last $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:[e]COVID-19 $[l]remains $infectious $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:flu $virus $persists $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:flu $virus $can $survive $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]flu $can $survive $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:flu $virus $[l]remains $viable $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:flu $virus $[l]can $last $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:flu $virus $[l]remains $infectious $on <E|U+C>arg2:surfaces\t47\n",
      "<E|U+C>arg1:[e]COVID-19 $persists $on surfaces $for <E|U+C>arg2:10 days\t48\n",
      "<E|U+C>arg1:[e]COVID-19 $can $survive $on surfaces $for <E|U+C>arg2:10 days\t48\n",
      "<E|U+C>arg1:[e]COVID-19 $[l]remains $viable $on surfaces for <E|U+C>arg2:10 days\t48\n",
      "<E|U+C>arg1:[e]COVID-19 $[l]can $last $on surfaces for <E|U+C>arg2:10 days\t48\n",
      "<E|U+C>arg1:[e]COVID-19 $[l]remains $infectious $on surfaces for <E|U+C>arg2:10 days\t48\n",
      "<E|U+C>arg1:flu $virus $persists $on surfaces for <E|U+C>arg2:10 days\t 48\n",
      "<E|U+C>arg1:flu $virus $can $survive $on surfaces for <E|U+C>arg2:10 days\t48\n",
      "<E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE|COVID-19]flu $can $survive $on surfaces for <E|U+C>arg2:10 days\t48\n",
      "<E|U+C>arg1:flu $virus $[l]remains $viable $on surfaces for <E|U+C>arg2:10 days\t48\n",
      "<E|U+C>arg1:flu $virus, which can $[l=persist|detect|live|survive]persist $on surfaces for <E|U+C>arg2:10 days\t48\n",
      "<E|U+C>arg1:flu $virus $[l]can $last $on surfaces for <E|U+C>arg2:10 days\t 48\n",
      "<E|U+C>arg1:flu $virus $[l]remains $infectious $on surfaces for <E|U+C>arg2:10 days\t48\n",
      "in <E|U+C>arg1:10% of the cases, <E|U+C>arg2:[e]COVID-19 patients $show $no $symptoms\t49\n",
      "in <E|U+C>arg1:10% of the cases, <E|U+C>arg2:[e]COVID-19 patients $have $no $symptoms\t49\n",
      "in <E|U+C>arg1:10% of the cases, <E|U+C>arg2:[e]COVID-19 patients $[l]do $not $show $symptoms\t49\n",
      "in <E|U+C>arg1:10% of <E|U+C>arg2:flu $infections the patients are $asymptomatic\t49\n",
      "in <E|U+C>arg1:10% of <E|U+C>arg2:flu $infections the patients $show $no $symptoms\t49\n",
      "in <E|U+C>arg1:10% of <E|U+C>arg2:flu $infections the patients $have $no $symptoms\t49\n",
      "in <E|U+C>arg1:10% of <E|U+C>arg2:flu $infections the patients $[l]do $not $show $symptoms\t49\n",
      "<E|U+C>arg1:20% of <E|U+C>arg2:flu $infections are $asymptomatic\t49\n",
      "<E|U+C>arg1:20 $percent of <E|U+C>arg2:flu $infections are $asymptomatic\t49\n",
      "<E|U+C>arg1:20 $percent of people with <E|U+C>arg2:flu $infections have $no $symptoms\t49\n",
      "the $proportion of $asymptomatic <E|U+C>arg2:[e]COVID-19 cases is <E|U+C>arg1:10%\t49\n",
      "<E|U+C>arg1:20% of people $infected $with <E|U+C>arg2:something $[l]are $asymptomatic\t49\n",
      "<E|U+C>arg1:something $is $synthesized $from <E|U+C>arg2:something\t50\n",
      "<E|U+C>arg1:something $is $a $derivative $of <E|U+C>arg2:something\t50\n",
      "<E|U+C>arg1:something $is $a $by $product $of <E|U+C>arg2:something $synthesis\t50\n",
      "<E|U+C>arg1:something $is $a $[l]by-product $of <E|U+C>arg2:something $synthesis\t50\n",
      "<E|U+C>arg1:something is $manufactured $from <E|U+C>arg2:something\t50\n",
      "$[w=synthesis|generation]synthesis of <E|U+C>arg1:something $from <E|U+C>arg2:something\t50\n",
      "<E|U+C>arg1:it $is $a <E|U+C>arg2:purine $derivative\t50\n",
      "the $conversion of <E|U+C>arg2:something $to <E|U+C>arg1:something\t50\n",
      "<E|U+C>arg1:something, a <E|U+C>arg2:purine $derivative\t50\n",
      "<E|U+C>arg1:something, a $derivative of <E|U+C>arg2:purine\t50\n",
      "<E|U+C>arg1:something, $synthesized $from <E|U+C>arg2:something\t50\n",
      "<E|U+C>arg1:[e=SIMPLE_CHEMICAL|CHEMICAL]something, which is $synthesized $from <E|U+C>arg2:something\t50\n",
      "<E|U+C>arg2:purine $[l]derivative $such $as <E|U+C>arg1:something\t50\n",
      "the <E|U+C>arg1:flu $[l]virus $[l]belongs $to $the $family $of <E|U+C>arg2:something\t51\n",
      "the <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]flu $[l]belongs $to $the $family $of <E|U+C>arg2:something\t51\n",
      "the <E|U+C>arg1:[e]COVID-19 $[l]belongs $to $the $family $of <E|U+C>arg2:something\t51\n",
      "$the $family $of <E|U+C>arg2:something that $[l=contain|include]contains <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBSTANCE|ORGANISM_SUBDIVISION]COVID-19\t51\n",
      "the <E|U+C>arg1:[e]MERS-CoV $[l]belongs $to $the $family $of <E|U+C>arg2:something\t51\n",
      "the <E|U+C>arg1:flu $bacteria $[l]belongs $to $the $family $of <E|U+C>arg2:something\t51\n",
      "the <E|U+C>arg1:flu $virus $is $a $part $of $the $family $of <E|U+C>arg2:something\t51\n",
      "the <E|U+C>arg1:[e]COVID-19 $is $a $part $of <E|U+C>arg2:something $family\t51\n",
      "the <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]COVID-19 $is $a $part $of <E|U+C>arg2:something $family\t51\n",
      "the <E|U+C>arg1:[e]MERS-CoV $is $a $part $of <E|U+C>arg2:something $family\t51\n",
      "the <E|U+C>arg1:flu $bacteria $is $a $part $of <E|U+C>arg2:something $family\t51\n",
      "the <E|U+C>arg1:[e]COVID-19 $is $a $part $of a $family of <E|U+C>arg2:something\t51\n",
      "the <E|U+C>arg1:[e]MERS-CoV $is $a $part $of a $family $of <E|U+C>arg2:something\t51\n",
      "the <E|U+C>arg1:flu $bacteria $is $a $part $of a $family $of <E|U+C>arg2:something\t51\n",
      "<E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE|COVID-19]flu $is $a $part $of a $family $of <E|U+C>arg2:something\t51\n",
      "the <E|U+C>arg1:flu $is $a $strain $of a <E|U+C>arg2:something\t51\n",
      "<E|U+C>arg1:flu, $a $strain $of <E|U+C>arg2:something\t51\n",
      "<E|U+C>arg1:something $is $a $parasite in <E|U+C>arg2:[e=ORGANISM|ORGAN|ORGANISM_SUBSTANCE]bats\t52\n",
      "<E|U+C>arg1:something, $a $parasite in <E|U+C>arg2:[e=ORGANISM|ORGAN|ORGANISM_SUBSTANCE]bats\t52\n",
      "<E|U+C>arg1:something $is $a $parasite $infecting <E|U+C>arg2:[e=ORGANISM|ORGAN|ORGANISM_SUBSTANCE]bats\t52\n",
      "<E|U+C>arg1:something, $a $parasite $infecting <E|U+C>arg2:[e=ORGANISM|ORGAN|ORGANISM_SUBSTANCE]bats\t52\n",
      "<E|U+C>arg1:something which is a $parasite $infecting <E|U+C>arg2:[e=ORGANISM|ORGAN|ORGANISM_SUBSTANCE]bats\t52\n",
      "<E|U+C>arg1:something $is $a <E|U+C>arg2:[e=ORGANISM|ORGAN|ORGANISM_SUBSTANCE]bats $parasite\t52\n",
      "<E|U+C>arg1:something, $a <E|U+C>arg2:[e]bats $parasite\t52\n",
      "<E|U+C>arg1:something $is $a $parasitic worm in <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]bats\t52\n",
      "<E|U+C>arg1:something, $a $parasitic worm in <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]bats\t52\n",
      "<E|U+C>arg1:something $is $a $parasitic worm $infecting <E|U+C>arg2:bats\t52\n",
      "<E|U+C>arg1:something, $a $parasitic worm $infecting <E|U+C>arg2:bats\t52\n",
      "<E|U+C>arg1:something $is $a $parasite $living in the lung of <E|U+C>arg2:[e]bats\t52\n",
      "<E|U+C>arg1:something, $a $parasite $living in the lung of <E|U+C>arg2:[e]bats\t52\n",
      "the <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]pig $parasite arg2:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]Anatidae\t52\n",
      "<E|U+C>arg1:[e=DISEASE]disease may $[l=require|necessitate|need]require <E|U+C>arg2:[e=ORGAN]lung $transplant\t53\n",
      "<E|U+C>arg1:[e=DISEASE]disease, that may $[l=require|necessitate|need]require <E|U+C>arg2:[e=ORGAN]lung $transplant\t53\n",
      "<E|U+C>arg1:[e=ORGAN]Liver $transplant is a $[l=treatment|measure|operation]treatment for <E|U+C>arg2:[e=DISEASE]flu\t53\n",
      "<E|U+C>arg1:[e=DISEASE]flu may $involve the <E|U+C>arg2:[e=ORGAN]liver\t54\n",
      "<E|U+C>arg1:[e=DISEASE]flu with <E|U+C>arg2:[e=ORGAN]liver $involvement\t54\n",
      "<E|U+C>arg2:[e=ORGAN]liver $involvement in <E|U+C>arg1:[e=DISEASE]flu\t54\n",
      "<E|U+C>arg1:[e=DISEASE]flu, which may $[l]involve the <E|U+C>arg2:[e=ORGAN]liver\t54\n",
      "<E|U+C>arg2:[e=ORGAN]liver may be affected from <E|U+C>arg1:[e=DISEASE]flu\t54\n",
      "<E|U+C>arg2:[e=ORGAN]liver, which may be $[l=affect|influence|damage]affected from <E|U+C>arg1:[e=DISEASE]flu\t54\n",
      "<E|U+C>arg1:[e=CELLULAR_COMPONENT]mitochondria is $responsible for <E|U+C>arg2:energy production\t55\n",
      "<E|U+C>arg1:[e=CELLULAR_COMPONENT]mitochondria, which is $responsible for <E|U+C>arg2:energy production\t55\n",
      "<E|U+C>arg2:energy $[l=produce|create|generate]produced by the <E|U+C>arg1:[e=CELLULAR_COMPONENT]mitochondria\t55\n",
      "<E|U+C>arg1:[e=CELL_LINE]something is used to $[l=simulate|study|develop|measure|propagate|investigate|detect|isolate]study <E|U+C>arg2:[e=DISEASE]something\t56\n",
      "<E|U+C>arg1:[e=CELL_LINE]something, which is used to $[l=simulate|study|develop|measure|propagate|investigate|detect|isolate]study <E|U+C>arg2:[e=DISEASE]something\t56\n",
      "<E|U+C>arg1:[e=DISEASE|ORGANISM|ORGANISM_SUBSTANCE]something was $[l=perform|simulate|show|use|observe|measure|examine|test|establish|determine|assess|make|diagnose|utilize|isolate]studied using <E|U+C>arg2:[e=CELL_LINE]something\t56\n",
      "<E|U+C>arg1:[e=DISEASE|ORGANISM|ORGANISM_SUBSTANCE]something, which was $[l=perform|simulate|show|use|observe|measure|examine|test|establish|determine|assess|make|diagnose|utilize|isolate]studied using <E|U+C>arg2:[e=CELL_LINE]something\t56\n",
      "<E|U+C>arg1:[e=DISEASE|ORGANISM|ORGANISM_SUBSTANCE]something linked to <E|U+C>arg2:[e=CELL_LINE]something\t56\n",
      "the $[w=examination|transduction|simulation|test|testing|determination|observation|evaluation]examination of <E|U+C>arg1:[e=DISEASE|ORGANISM|ORGANISM_SUBSTANCE]something $[w=with|using|by]with <E|U+C>arg2:[e=CELL_LINE]something\t56\n",
      "it is $[word=unknown|unclear]unknown how <E|U+C>arg1:something affects <E|U+C>arg2:something\t57\n",
      "$future $research is needed to examine how <E|U+C>arg1:something affects <E|U+C>arg2:something\t57\n",
      "the way <E|U+C>arg1:something affects <E|U+C>arg2:something remains $elusive\t57\n",
      "<E|U+C>arg1:something effect on <E|U+C>arg2:something remains $elusive\t57\n",
      "<E|U+C>arg1:something $[w=can|should]can $not $be $attributed to <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]something\t58\n",
      "<E|U+C>arg1:something $is $not $due to <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]something\t58\n",
      "<E|U+C>arg1:[e=DISEASE]something $not $due $to <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]something\t58\n",
      "<E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]something $is $not $responsible for <E|U+C>arg1:[e=DISEASE]something\t58\n",
      "<E|U+C>arg1:[e=DISEASE]something is a $differential $diagnosis for <E|U+C>arg2:[e=DISEASE]something\t59\n",
      "<E|U+C>arg1:[e=DISEASE]something $should be $differentiated from <E|U+C>arg2:[e=DISEASE]something\t59\n",
      "<E|U+C>arg1:[e=DISEASE]something, a $differential diagnosis for <E|U+C>arg2:[e=DISEASE]something\t59\n",
      "<E|U+C>arg1:[e=DISEASE|COVID-19]something $can $mimic <E|U+C>arg2:[e=DISEASE|COVID-19]something\t59\n",
      "<E|U+C>arg1:[e=DISEASE]something $misdiagnosed $as <E|U+C>arg2:[e=DISEASE]something\t59\n",
      "<E|U+C>arg1:[e=DISEASE]something is $misdiagnosed $as <E|U+C>arg2:[e=DISEASE]something\t59\n",
      "<E|U+C>arg1:[e=DISEASE]something, which is $misdiagnosed $as <E|U+C>arg2:[e=DISEASE]something\t59\n",
      "the $misdiagnosis of <>arg1:[e=DISEASE]something $as <>arg2:[e=DISEASE]something\t59\n",
      "the $differential $diagnosis of <E|U+C>arg1:[e=DISEASE]something $from <E|U+C>arg2:[e=DISEASE]something\t59\n",
      "<E|U+C>arg1:[e=DISEASE]something $confused with <E|U+C>arg2:[e=DISEASE]something\t59\n",
      "<E|U+C>arg1:[e=DISEASE]something $is $confused with <E|U+C>arg2:[e=DISEASE]something\t59\n",
      "the $differentiation of <E|U+C>arg1:[e=DISEASE]something $from <E|U+C>arg2:[e=DISEASE]something\t59\n",
      "<E|U+C>arg1:[e=DISEASE|ORGANISM_SUBSTANCE|ORGANISM|COVID-19]flu :[lemma=cause|cost]causes <E|U+C>arg2:something $$ \t60\n",
      "the :[lemma=cost|price|toll]cost of <E|U+C>arg1:[e=DISEASE|ORGANISM_SUBSTANCE|ORGANISM|COVID-19]flu is <E|U+C>arg2:something $$\t60\n",
      "<E|U+C>arg1:[e=DISEASE|ORGANISM_SUBSTANCE|ORGANISM|COVID-19]flu, which :[lemma=cause|cost]costs <E|U+C>arg2:something $$\t60\n",
      "<E|U+C>arg1:[e=DISEASE|ORGANISM_SUBSTANCE|ORGANISM|COVID-19]flu :[lemma=cause|cost]causes <E|U+C>arg2:something $[w=dollar|USD]dollar \t60\n",
      "the :[lemma=cost|price|toll]cost of <E|U+C>arg1:[e=DISEASE|ORGANISM_SUBSTANCE|ORGANISM|COVID-19]flu is <E|U+C>arg2:something $[w=dollar|USD]dollar\t60\n",
      "<E|U+C>arg1:[e=DISEASE|ORGANISM_SUBSTANCE|ORGANISM|COVID-19]flu, which :[lemma=cause|cost]costs <E|U+C>arg2:something $[w=dollar|USD]dollar\t60\n",
      "<E|U+C>arg1:something is a arg2:plant $derived $[l=drug|medicine|agent]drug\t61\n",
      "<E|U+C>arg1:something is a $[l=drug|medicine|agent]drug $[l=extract|distill|obtain|derive]extracted from <E|U+C>arg2:plant\t61\n",
      "<E|U+C>arg1:something, a <E|U+C>arg2:plant $derived $[l=drug|medicine|agent]drug\t61\n",
      "<E|U+C>arg1:something, which is a $[l=drug|medicine|agent]drug $[l=extract|distill|obtain]extracted from <E|U+C>arg2:something\t61\n",
      "<E|U+C>arg1:something, a $[l=drug|medicine|agent]drug $[l=extract|distill|obtain]extracted from <E|U+C>arg2:something\t61\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]drug, a $[l=drug|medicine|agent|molecule]drug extracted $from <E|U+C>arg2:[l=plant|yeast|bacteria|soil|fungus|fungi|bark]plants\t61\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]drug is a $[l=drug|medicine|agent|molecule]drug extracted $from <E|U+C>arg2:[l=plant|yeast|bacteria|soil|fungus|fungi|bark|leaf]plants\t61\n",
      "['<E|U+C>arg1:virus $infection $causes a <E|U+C>arg2:condition', '<E|U+C>arg1:virus $infection, which $causes a <E|U+C>arg2:condition', '<E|U+C>arg1:virus $infection $results $in <E|U+C>arg2:condition', '<E|U+C>arg1:virus $infection, which $results $in <E|U+C>arg2:condition', '<E|U+C>arg2:condition $is $the $result of <E|U+C>arg1:virus $infection', '<E|U+C>arg2:condition, which $is $the $result of <E|U+C>arg1:virus $infection', '<E|U+C>arg2:condition $is $[l=cause|trigger]caused $by <E|U+C>arg1:virus $infection', 'as a result of <E|U+C>arg1:virus $infection, <E|U+C>arg2:condition $can $develop']\n"
     ]
    }
   ],
   "source": [
    "queries2results = defaultdict(list)\n",
    "id2query = defaultdict(list)\n",
    "\n",
    "for id_and_query in queries:\n",
    "    print(id_and_query)\n",
    "    query,id = id_and_query.split(\"\\t\")\n",
    "    id2query[int(id)].append(query)\n",
    "    \n",
    "print(id2query[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 422/422 [1:00:25<00:00,  8.59s/it]\n"
     ]
    }
   ],
   "source": [
    "# dataset_name = \"covid19\"\n",
    "# num_results = 150\n",
    "# query_type = \"syntactic\"\n",
    "# import time\n",
    "\n",
    "# for i, q_and_id in tqdm.tqdm(enumerate(queries), total = len(queries)):\n",
    "#     q, id = q_and_id.split(\"\\t\")\n",
    "#     id = int(id)\n",
    "#     try:\n",
    "#         df = spike_queries.perform_query(q, dataset_name, num_results, query_type) #previously: word=Hawaii\n",
    "#         queries2results[id].append(df)\n",
    "#         time.sleep(2)\n",
    "#     except Exception as e:\n",
    "#         print(\"Error\", i+1)\n",
    "#         print(e)\n",
    "        \n",
    "#         continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 0\n",
      "1 1\n",
      "1 2\n",
      "1 3\n",
      "1 4\n",
      "1 5\n",
      "1 6\n",
      "1 7\n",
      "2 0\n",
      "2 1\n",
      "2 2\n",
      "3 0\n",
      "3 1\n",
      "4 0\n",
      "4 1\n",
      "4 2\n",
      "4 3\n",
      "4 4\n",
      "4 5\n",
      "4 6\n",
      "4 7\n",
      "5 0\n",
      "5 1\n",
      "5 2\n",
      "5 3\n",
      "5 4\n",
      "5 5\n",
      "5 6\n",
      "5 7\n",
      "5 8\n",
      "5 9\n",
      "5 10\n",
      "6 0\n",
      "6 1\n",
      "6 2\n",
      "6 3\n",
      "6 4\n",
      "6 5\n",
      "6 6\n",
      "6 7\n",
      "6 8\n",
      "7 0\n",
      "7 1\n",
      "7 2\n",
      "7 3\n",
      "7 4\n",
      "8 0\n",
      "8 1\n",
      "8 2\n",
      "8 3\n",
      "9 0\n",
      "9 1\n",
      "9 2\n",
      "9 3\n",
      "9 4\n",
      "9 5\n",
      "9 6\n",
      "9 7\n",
      "9 8\n",
      "10 0\n",
      "10 1\n",
      "10 2\n",
      "10 3\n",
      "10 4\n",
      "10 5\n",
      "10 6\n",
      "10 7\n",
      "11 0\n",
      "11 1\n",
      "11 2\n",
      "11 3\n",
      "11 4\n",
      "11 5\n",
      "11 6\n",
      "11 7\n",
      "11 8\n",
      "11 9\n",
      "11 10\n",
      "12 0\n",
      "12 1\n",
      "12 2\n",
      "12 3\n",
      "13 0\n",
      "13 1\n",
      "13 2\n",
      "13 3\n",
      "13 4\n",
      "14 0\n",
      "14 1\n",
      "14 2\n",
      "14 3\n",
      "14 4\n",
      "14 5\n",
      "14 6\n",
      "15 0\n",
      "15 1\n",
      "15 2\n",
      "15 3\n",
      "15 4\n",
      "15 5\n",
      "15 6\n",
      "15 7\n",
      "15 8\n",
      "15 9\n",
      "15 10\n",
      "15 11\n",
      "15 12\n",
      "16 0\n",
      "16 1\n",
      "16 2\n",
      "16 3\n",
      "16 4\n",
      "16 5\n",
      "16 6\n",
      "16 7\n",
      "16 8\n",
      "17 0\n",
      "17 1\n",
      "17 2\n",
      "17 3\n",
      "18 0\n",
      "18 1\n",
      "18 2\n",
      "18 3\n",
      "18 4\n",
      "18 5\n",
      "19 0\n",
      "19 1\n",
      "19 2\n",
      "19 3\n",
      "19 4\n",
      "19 5\n",
      "19 6\n",
      "20 0\n",
      "20 1\n",
      "20 2\n",
      "20 3\n",
      "20 4\n",
      "21 0\n",
      "21 1\n",
      "21 2\n",
      "21 3\n",
      "21 4\n",
      "21 5\n",
      "22 0\n",
      "22 1\n",
      "22 2\n",
      "22 3\n",
      "22 4\n",
      "22 5\n",
      "23 0\n",
      "23 1\n",
      "23 2\n",
      "23 3\n",
      "23 4\n",
      "23 5\n",
      "23 6\n",
      "23 7\n",
      "23 8\n",
      "23 9\n",
      "23 10\n",
      "24 0\n",
      "24 1\n",
      "24 2\n",
      "25 0\n",
      "25 1\n",
      "25 2\n",
      "25 3\n",
      "26 0\n",
      "26 1\n",
      "26 2\n",
      "27 0\n",
      "27 1\n",
      "27 2\n",
      "28 0\n",
      "28 1\n",
      "28 2\n",
      "29 0\n",
      "29 1\n",
      "29 2\n",
      "29 3\n",
      "29 4\n",
      "29 5\n",
      "29 6\n",
      "30 0\n",
      "30 1\n",
      "30 2\n",
      "30 3\n",
      "30 4\n",
      "31 0\n",
      "31 1\n",
      "31 2\n",
      "31 3\n",
      "31 4\n",
      "31 5\n",
      "31 6\n",
      "32 0\n",
      "32 1\n",
      "32 2\n",
      "32 3\n",
      "32 4\n",
      "32 5\n",
      "32 6\n",
      "33 0\n",
      "33 1\n",
      "33 2\n",
      "33 3\n",
      "34 0\n",
      "34 1\n",
      "34 2\n",
      "35 0\n",
      "35 1\n",
      "35 2\n",
      "35 3\n",
      "35 4\n",
      "35 5\n",
      "35 6\n",
      "35 7\n",
      "35 8\n",
      "35 9\n",
      "35 10\n",
      "35 11\n",
      "35 12\n",
      "35 13\n",
      "35 14\n",
      "35 15\n",
      "35 16\n",
      "35 17\n",
      "35 18\n",
      "36 0\n",
      "36 1\n",
      "36 2\n",
      "37 0\n",
      "37 1\n",
      "37 2\n",
      "37 3\n",
      "37 4\n",
      "37 5\n",
      "37 6\n",
      "37 7\n",
      "37 8\n",
      "38 0\n",
      "38 1\n",
      "38 2\n",
      "38 3\n",
      "38 4\n",
      "38 5\n",
      "38 6\n",
      "38 7\n",
      "38 8\n",
      "38 9\n",
      "39 0\n",
      "39 1\n",
      "39 2\n",
      "39 3\n",
      "39 4\n",
      "40 0\n",
      "40 1\n",
      "40 2\n",
      "40 3\n",
      "40 4\n",
      "40 5\n",
      "40 6\n",
      "40 7\n",
      "41 0\n",
      "41 1\n",
      "41 2\n",
      "42 0\n",
      "42 1\n",
      "42 2\n",
      "43 0\n",
      "43 1\n",
      "43 2\n",
      "43 3\n",
      "43 4\n",
      "43 5\n",
      "44 0\n",
      "44 1\n",
      "44 2\n",
      "44 3\n",
      "45 0\n",
      "45 1\n",
      "45 2\n",
      "45 3\n",
      "45 4\n",
      "46 0\n",
      "46 1\n",
      "46 2\n",
      "46 3\n",
      "46 4\n",
      "46 5\n",
      "46 6\n",
      "47 0\n",
      "47 1\n",
      "47 2\n",
      "47 3\n",
      "47 4\n",
      "47 5\n",
      "47 6\n",
      "47 7\n",
      "47 8\n",
      "47 9\n",
      "47 10\n",
      "48 0\n",
      "48 1\n",
      "48 2\n",
      "48 3\n",
      "48 4\n",
      "48 5\n",
      "48 6\n",
      "48 7\n",
      "48 8\n",
      "48 9\n",
      "48 10\n",
      "48 11\n",
      "49 0\n",
      "49 1\n",
      "49 2\n",
      "49 3\n",
      "49 4\n",
      "49 5\n",
      "49 6\n",
      "49 7\n",
      "49 8\n",
      "49 9\n",
      "49 10\n",
      "49 11\n",
      "50 0\n",
      "50 1\n",
      "50 2\n",
      "50 3\n",
      "50 4\n",
      "50 5\n",
      "50 6\n",
      "50 7\n",
      "50 8\n",
      "50 9\n",
      "50 10\n",
      "50 11\n",
      "50 12\n",
      "51 0\n",
      "51 1\n",
      "51 2\n",
      "51 3\n",
      "51 4\n",
      "51 5\n",
      "51 6\n",
      "51 7\n",
      "51 8\n",
      "51 9\n",
      "51 10\n",
      "51 11\n",
      "51 12\n",
      "51 13\n",
      "51 14\n",
      "51 15\n",
      "51 16\n",
      "52 0\n",
      "52 1\n",
      "52 2\n",
      "52 3\n",
      "52 4\n",
      "52 5\n",
      "52 6\n",
      "52 7\n",
      "52 8\n",
      "52 9\n",
      "52 10\n",
      "52 11\n",
      "52 12\n",
      "52 13\n",
      "53 0\n",
      "53 1\n",
      "53 2\n",
      "54 0\n",
      "54 1\n",
      "54 2\n",
      "54 3\n",
      "54 4\n",
      "54 5\n",
      "55 0\n",
      "55 1\n",
      "55 2\n",
      "56 0\n",
      "56 1\n",
      "56 2\n",
      "56 3\n",
      "56 4\n",
      "56 5\n",
      "57 0\n",
      "57 1\n",
      "57 2\n",
      "57 3\n",
      "58 0\n",
      "58 1\n",
      "58 2\n",
      "58 3\n",
      "59 0\n",
      "59 1\n",
      "59 2\n",
      "59 3\n",
      "59 4\n",
      "59 5\n",
      "59 6\n",
      "59 7\n",
      "59 8\n",
      "59 9\n",
      "59 10\n",
      "59 11\n",
      "60 0\n",
      "60 1\n",
      "60 2\n",
      "60 3\n",
      "60 4\n",
      "60 5\n",
      "61 0\n",
      "61 1\n",
      "61 2\n",
      "61 3\n",
      "61 4\n",
      "61 5\n",
      "61 6\n"
     ]
    }
   ],
   "source": [
    "for id, dfs in queries2results.items():\n",
    "     for i,df in enumerate(dfs):\n",
    "         print(id, i)\n",
    "         queries2results[id][i] = df.dropna(subset=['arg1_first_index',\"arg1_last_index\",\"arg2_first_index\",\"arg2_last_index\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "for id, dfs in queries2results.items():\n",
    "      for i,df in enumerate(dfs):\n",
    "            queries2results[id][i][\"spike_query\"] = id2query[id][i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"queries2results2.pickle\", \"wb\") as f:\n",
    "      pickle.dump(queries2results, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"queries2results2.pickle\", \"rb\") as f:\n",
    "    queries2results = pickle.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence_id</th>\n",
       "      <th>arg1</th>\n",
       "      <th>arg2</th>\n",
       "      <th>title</th>\n",
       "      <th>article_link</th>\n",
       "      <th>arg1_first_index</th>\n",
       "      <th>arg1_last_index</th>\n",
       "      <th>arg2_first_index</th>\n",
       "      <th>arg2_last_index</th>\n",
       "      <th>sentence_text</th>\n",
       "      <th>paragraph_text</th>\n",
       "      <th>spike_query</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11709</td>\n",
       "      <td>The virus infection</td>\n",
       "      <td>generalized infections</td>\n",
       "      <td>Speaker Presentations</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>The virus infection causes generalized infecti...</td>\n",
       "      <td>The co-emergence of multiple CA-MRSA lineages ...</td>\n",
       "      <td>&lt;E|U+C&gt;arg1:virus $infection $causes a &lt;E|U+C&gt;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>16550</td>\n",
       "      <td>Middle East respiratory syndrome coronavirus (...</td>\n",
       "      <td>an acute respiratory illness</td>\n",
       "      <td>Clinicopathologic, Immunohistochemical, and Ul...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>13</td>\n",
       "      <td>Middle East respiratory syndrome coronavirus (...</td>\n",
       "      <td>Middle East respiratory syndrome coronavirus (...</td>\n",
       "      <td>&lt;E|U+C&gt;arg1:virus $infection $causes a &lt;E|U+C&gt;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>42944</td>\n",
       "      <td>MHV infection</td>\n",
       "      <td>changes</td>\n",
       "      <td>viruses Of Mice and Men: The Coronavirus MHV a...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>MHV infection increases iron uptake due to tis...</td>\n",
       "      <td>MHV causes hepatitis, which results in elevate...</td>\n",
       "      <td>&lt;E|U+C&gt;arg1:virus $infection $causes a &lt;E|U+C&gt;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>51830</td>\n",
       "      <td>ZIKV infection</td>\n",
       "      <td>mild and self-limiting disease</td>\n",
       "      <td>ARTICLE IN PRESS</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>7 , 8 ZIKV infection usually causes mild and s...</td>\n",
       "      <td>The Flavivirus genus comprises many viruses di...</td>\n",
       "      <td>&lt;E|U+C&gt;arg1:virus $infection $causes a &lt;E|U+C&gt;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>54089</td>\n",
       "      <td>SARS-CoV-2 infection</td>\n",
       "      <td>a relatively low release of pro-inflammatory c...</td>\n",
       "      <td>viruses Of Mice and Men: The Coronavirus MHV a...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>25</td>\n",
       "      <td>33</td>\n",
       "      <td>A possible reason for this is the suppression ...</td>\n",
       "      <td>SARS-CoV-2 replicates more efficiently than SA...</td>\n",
       "      <td>&lt;E|U+C&gt;arg1:virus $infection $causes a &lt;E|U+C&gt;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>2477435</td>\n",
       "      <td>Fetal cytomegalovirus infection</td>\n",
       "      <td>neurologic damage</td>\n",
       "      <td>Editorial Addressing the Effects of Establishe...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>Fetal cytomegalovirus infection causes neurolo...</td>\n",
       "      <td>Also in this issue is an update by Davis et al...</td>\n",
       "      <td>&lt;E|U+C&gt;arg1:virus $infection $causes a &lt;E|U+C&gt;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>2497840</td>\n",
       "      <td>SARS-CoV infection</td>\n",
       "      <td>severe symptoms</td>\n",
       "      <td>Pathogenesis of SARS 1145</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>1 , 3 In contrast with previously described co...</td>\n",
       "      <td>A novel coronavirus was identified as the etio...</td>\n",
       "      <td>&lt;E|U+C&gt;arg1:virus $infection $causes a &lt;E|U+C&gt;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>2525863</td>\n",
       "      <td>ZIKV infection</td>\n",
       "      <td>placental pathology , fetal growth restriction...</td>\n",
       "      <td>Animal models of congenital zika syndrome prov...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>17</td>\n",
       "      <td>ZIKV infection during the first trimester also...</td>\n",
       "      <td>Sheep are a less recognized model of CZS, whic...</td>\n",
       "      <td>&lt;E|U+C&gt;arg1:virus $infection $causes a &lt;E|U+C&gt;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>2539632</td>\n",
       "      <td>ZIKV infection</td>\n",
       "      <td>reduced oxygen permeability of the placental v...</td>\n",
       "      <td>Animal models of congenital zika syndrome prov...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>18</td>\n",
       "      <td>In rhesus macaques , ZIKV infection ( Cambodia...</td>\n",
       "      <td>Rhesus macaques are the most widely utilized N...</td>\n",
       "      <td>&lt;E|U+C&gt;arg1:virus $infection $causes a &lt;E|U+C&gt;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>2539638</td>\n",
       "      <td>ZIKV infection of pregnant rhesus macaques and...</td>\n",
       "      <td>severe placental damage</td>\n",
       "      <td>Animal models of congenital zika syndrome prov...</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>ZIKV infection of pregnant rhesus macaques and...</td>\n",
       "      <td>Rhesus macaques are the most widely utilized N...</td>\n",
       "      <td>&lt;E|U+C&gt;arg1:virus $infection $causes a &lt;E|U+C&gt;...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>150 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     sentence_id                                               arg1  \\\n",
       "0          11709                                The virus infection   \n",
       "1          16550  Middle East respiratory syndrome coronavirus (...   \n",
       "2          42944                                      MHV infection   \n",
       "3          51830                                     ZIKV infection   \n",
       "4          54089                               SARS-CoV-2 infection   \n",
       "..           ...                                                ...   \n",
       "145      2477435                    Fetal cytomegalovirus infection   \n",
       "146      2497840                                 SARS-CoV infection   \n",
       "147      2525863                                     ZIKV infection   \n",
       "148      2539632                                     ZIKV infection   \n",
       "149      2539638  ZIKV infection of pregnant rhesus macaques and...   \n",
       "\n",
       "                                                  arg2  \\\n",
       "0                               generalized infections   \n",
       "1                         an acute respiratory illness   \n",
       "2                                              changes   \n",
       "3                       mild and self-limiting disease   \n",
       "4    a relatively low release of pro-inflammatory c...   \n",
       "..                                                 ...   \n",
       "145                                  neurologic damage   \n",
       "146                                    severe symptoms   \n",
       "147  placental pathology , fetal growth restriction...   \n",
       "148  reduced oxygen permeability of the placental v...   \n",
       "149                            severe placental damage   \n",
       "\n",
       "                                                 title  \\\n",
       "0                                Speaker Presentations   \n",
       "1    Clinicopathologic, Immunohistochemical, and Ul...   \n",
       "2    viruses Of Mice and Men: The Coronavirus MHV a...   \n",
       "3                                     ARTICLE IN PRESS   \n",
       "4    viruses Of Mice and Men: The Coronavirus MHV a...   \n",
       "..                                                 ...   \n",
       "145  Editorial Addressing the Effects of Establishe...   \n",
       "146                          Pathogenesis of SARS 1145   \n",
       "147  Animal models of congenital zika syndrome prov...   \n",
       "148  Animal models of congenital zika syndrome prov...   \n",
       "149  Animal models of congenital zika syndrome prov...   \n",
       "\n",
       "                                          article_link  arg1_first_index  \\\n",
       "0    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                 0   \n",
       "1    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                 0   \n",
       "2    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                 0   \n",
       "3    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                 3   \n",
       "4    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                14   \n",
       "..                                                 ...               ...   \n",
       "145  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                 0   \n",
       "146  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1...                10   \n",
       "147  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                 0   \n",
       "148  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                 4   \n",
       "149  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...                 0   \n",
       "\n",
       "     arg1_last_index  arg2_first_index  arg2_last_index  \\\n",
       "0                  2                 4                5   \n",
       "1                  8                10               13   \n",
       "2                  1                17               17   \n",
       "3                  4                 7               10   \n",
       "4                 15                25               33   \n",
       "..               ...               ...              ...   \n",
       "145                2                 4                5   \n",
       "146               11                14               15   \n",
       "147                1                 8               17   \n",
       "148                5                12               18   \n",
       "149                7                10               12   \n",
       "\n",
       "                                         sentence_text  \\\n",
       "0    The virus infection causes generalized infecti...   \n",
       "1    Middle East respiratory syndrome coronavirus (...   \n",
       "2    MHV infection increases iron uptake due to tis...   \n",
       "3    7 , 8 ZIKV infection usually causes mild and s...   \n",
       "4    A possible reason for this is the suppression ...   \n",
       "..                                                 ...   \n",
       "145  Fetal cytomegalovirus infection causes neurolo...   \n",
       "146  1 , 3 In contrast with previously described co...   \n",
       "147  ZIKV infection during the first trimester also...   \n",
       "148  In rhesus macaques , ZIKV infection ( Cambodia...   \n",
       "149  ZIKV infection of pregnant rhesus macaques and...   \n",
       "\n",
       "                                        paragraph_text  \\\n",
       "0    The co-emergence of multiple CA-MRSA lineages ...   \n",
       "1    Middle East respiratory syndrome coronavirus (...   \n",
       "2    MHV causes hepatitis, which results in elevate...   \n",
       "3    The Flavivirus genus comprises many viruses di...   \n",
       "4    SARS-CoV-2 replicates more efficiently than SA...   \n",
       "..                                                 ...   \n",
       "145  Also in this issue is an update by Davis et al...   \n",
       "146  A novel coronavirus was identified as the etio...   \n",
       "147  Sheep are a less recognized model of CZS, whic...   \n",
       "148  Rhesus macaques are the most widely utilized N...   \n",
       "149  Rhesus macaques are the most widely utilized N...   \n",
       "\n",
       "                                           spike_query  \n",
       "0    <E|U+C>arg1:virus $infection $causes a <E|U+C>...  \n",
       "1    <E|U+C>arg1:virus $infection $causes a <E|U+C>...  \n",
       "2    <E|U+C>arg1:virus $infection $causes a <E|U+C>...  \n",
       "3    <E|U+C>arg1:virus $infection $causes a <E|U+C>...  \n",
       "4    <E|U+C>arg1:virus $infection $causes a <E|U+C>...  \n",
       "..                                                 ...  \n",
       "145  <E|U+C>arg1:virus $infection $causes a <E|U+C>...  \n",
       "146  <E|U+C>arg1:virus $infection $causes a <E|U+C>...  \n",
       "147  <E|U+C>arg1:virus $infection $causes a <E|U+C>...  \n",
       "148  <E|U+C>arg1:virus $infection $causes a <E|U+C>...  \n",
       "149  <E|U+C>arg1:virus $infection $causes a <E|U+C>...  \n",
       "\n",
       "[150 rows x 12 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "queries2results[1][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# queries2results_new = dict()\n",
    "# for k,dfs in queries2results.items():\n",
    "#     if k == 0: continue\n",
    "#     queries2results_new[k] = pd.concat(dfs, axis=0)\n",
    "    \n",
    "# queries2results = queries2results_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "ids = set([int(l.split(\"\\t\")[-1]) for l in queries])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_arguments(sent:str, arg1_start, arg1_end, arg2_start, arg2_end):\n",
    "    \n",
    "        s_lst = sent.split(\" \")\n",
    "        if arg1_start > arg2_start:\n",
    "            arg1_start, arg2_start = arg2_start, arg1_start\n",
    "            arg1_end, arg2_end = arg2_end, arg1_end\n",
    "            arg1_str, arg2_str = \"{{ARG2:\", \"<<ARG1:\"\n",
    "        else:\n",
    "            arg1_str, arg2_str = \"<<ARG1:\", \"{{ARG2:\"\n",
    "        \n",
    "        s_with_args = s_lst[:arg1_start] + [arg1_str] + s_lst[arg1_start:arg1_end+1] + [\">>\"] + s_lst[arg1_end+1:arg2_start] + [arg2_str] + s_lst[arg2_start:arg2_end+1] + [\"}}\"] +s_lst[arg2_end+1:]  \n",
    "        #s_with_args = s_lst[:arg1_start] + [arg1_str+s_lst[arg1_ind]] + s_lst[arg1_ind+1:arg2_ind] + [arg2_str+s_lst[arg2_ind]] + s_lst[arg2_ind+1:]\n",
    "        s_with_args = \" \".join(s_with_args).replace(\"ARG1: \", \"ARG1:\").replace(\"ARG2: \", \"ARG2:\")\n",
    "        s_with_args = s_with_args.replace(\" >>\", \">>\").replace(\" }}\", \"}}\")\n",
    "        return s_with_args\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 61/61 [00:18<00:00,  3.34it/s]\n"
     ]
    }
   ],
   "source": [
    "NUM_PAIRS_PER_RELATION = 1400\n",
    "\n",
    "data = []\n",
    "\n",
    "for id in tqdm.tqdm(ids): #foreach relation\n",
    "    \n",
    "    pattern_pairs = list(itertools.product(range(len(id2query[id])), repeat=2))\n",
    "    pattern_pairs_different = [(p1,p2) for (p1,p2) in pattern_pairs if p1 != p2]\n",
    "    pattern_pairs_same = [(p1,p2) for (p1,p2) in pattern_pairs if p1 == p2]\n",
    "    pattern_pairs_different += pattern_pairs_same\n",
    "    sum_pattern = 0\n",
    "    relation_examples = []\n",
    "    \n",
    "    for p1, p2 in pattern_pairs_different:\n",
    "        df1,df2 = queries2results[id][p1], queries2results[id][p2]\n",
    "        if df1.empty or df2.empty: continue\n",
    "            \n",
    "        sentences1, sentences2 = df1[\"sentence_text\"].tolist(), df2[\"sentence_text\"].tolist()\n",
    "        query1, query2 = df1[\"spike_query\"].tolist()[0], df2[\"spike_query\"].tolist()[0]\n",
    "        arg1_first_start, arg1_first_end = df1[\"arg1_first_index\"].tolist(), df1[\"arg1_last_index\"].tolist()\n",
    "        arg2_first_start, arg2_first_end = df1[\"arg2_first_index\"].tolist(), df1[\"arg2_last_index\"].tolist()\n",
    "        arg1_second_start, arg1_second_end = df2[\"arg1_first_index\"].tolist(), df2[\"arg1_last_index\"].tolist()\n",
    "        arg2_second_start, arg2_second_end = df2[\"arg2_first_index\"].tolist(), df2[\"arg2_last_index\"].tolist()\n",
    "        \n",
    "        \n",
    "        all_pair_combinations = list(itertools.product(range(len(df1)), range(len(df2))))\n",
    "        random.shuffle(all_pair_combinations)\n",
    "        for combination in all_pair_combinations[:100]:\n",
    "            ind1, ind2 = combination\n",
    "            \n",
    "            sent1, sent2 = sentences1[ind1], sentences2[ind2]\n",
    "            sent1_arg1_start, sent1_arg1_end = arg1_first_start[ind1], arg1_first_end[ind1]\n",
    "            sent1_arg2_start, sent1_arg2_end = arg2_first_start[ind1], arg2_first_end[ind1]\n",
    "            sent2_arg1_start, sent2_arg1_end = arg1_second_start[ind2], arg1_second_end[ind2]\n",
    "            sent2_arg2_start, sent2_arg2_end = arg2_second_start[ind2], arg2_second_end[ind2]\n",
    "            sent1_with_args = add_arguments(sent1, sent1_arg1_start, sent1_arg1_end, sent1_arg2_start, sent1_arg2_end)\n",
    "            sent2_with_args = add_arguments(sent2, sent2_arg1_start, sent2_arg1_end, sent2_arg2_start, sent2_arg2_end)\n",
    "            \n",
    "            sent1_lst = sent1.split(\" \")\n",
    "            sent2_lst = sent2.split(\" \")\n",
    "            \n",
    "            sent1_arg1_w = sent1_lst[sent1_arg1_start:sent1_arg1_end+1]\n",
    "            sent1_arg2_w = sent1_lst[sent1_arg2_start:sent1_arg2_end+1]\n",
    "            sent2_arg1_w = sent2_lst[sent2_arg1_start:sent2_arg1_end+1]\n",
    "            sent2_arg2_w = sent2_lst[sent2_arg2_start:sent2_arg2_end+1]\n",
    "            \n",
    "            d = {\"first\": sent1_with_args, \"second\": sent2, \"second_with_arguments\": sent2_with_args, \"query_first\": query1, \"query_second\": query2, \n",
    "                 \"first_arg1\": (sent1_arg1_start, sent1_arg1_end), \"first_arg2\": (sent1_arg2_start, sent1_arg2_end), \n",
    "                 \"second_arg1\": (sent2_arg1_start, sent2_arg1_end),\n",
    "                 \"second_arg2\": (sent2_arg2_start, sent2_arg2_end), \"first_arg1_words\": sent1_arg2_w,\n",
    "                \"first_arg2_words\": sent1_arg2_w, \"second_arg1_words\": sent2_arg1_w, \"second_arg2_words\": sent2_arg2_w,\n",
    "                \"query_id\": id}\n",
    "            relation_examples.append(d)\n",
    "    \n",
    "    random.shuffle(relation_examples)\n",
    "    relation_examples = relation_examples[:NUM_PAIRS_PER_RELATION]\n",
    "    data.extend(relation_examples)\n",
    "    \n",
    "random.shuffle(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'first': 'O. pedunculosum contains <<ARG1:3-O-methylquercetin glucosides>> that inhibit lipopolysaccharideinduced NO production in {{ARG2:macrophages}} ( Wan et al. 2012 ) , and a mannan-specific lectin with antifungal activity ( He et al. 2011 ) .',\n",
       " 'second': 'However , they can do even better than that : Prof. Clemen Posten and his team at the Karlsruhe Institute of Technology are working on bioprocesses that enable them to cultivate the green alga Chlamydomonas reinhardtii in a way that triggers it to produce environmentallyfriendly hydrogen , both more cheaply and more energy efficiently than has been possible up until now .',\n",
       " 'second_with_arguments': 'However , they can do even better than that : Prof. Clemen Posten and his team at the Karlsruhe Institute of Technology are working on bioprocesses that enable them to cultivate the green alga Chlamydomonas reinhardtii in <<ARG1:a way>> that triggers it to produce environmentallyfriendly {{ARG2:hydrogen}} , both more cheaply and more energy efficiently than has been possible up until now .',\n",
       " 'query_first': '<E|U+C>arg1:something, that $[l=encourage|trigger|inhibit]encourages the $production of <E|U+C>arg2:something',\n",
       " 'query_second': '<E|U+C>arg1:something $[l=trigger|encourage]triggers the body $to $produce <E|U+C>arg2:something',\n",
       " 'first_arg1': (3, 4),\n",
       " 'first_arg2': (11, 11),\n",
       " 'second_arg1': (37, 38),\n",
       " 'second_arg2': (45, 45),\n",
       " 'first_arg1_words': ['macrophages'],\n",
       " 'first_arg2_words': ['macrophages'],\n",
       " 'second_arg1_words': ['a', 'way'],\n",
       " 'second_arg2_words': ['hydrogen'],\n",
       " 'query_id': 24}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*** Add negative examples ***"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[27 36 61 29 12  3 35 60 41 23  5 11]\n"
     ]
    }
   ],
   "source": [
    "lengths = [len(d[\"first\"].split(\" \")) + len(d[\"second\"].split(\" \")) for d in data]\n",
    "data = [d for i,d in enumerate(data) if lengths[i] < 250]\n",
    "data = [d for d in data if (d[\"second_arg1\"] != d[\"second_arg2\"]) and (d[\"first_arg1\"] != d[\"first_arg2\"])]\n",
    "data = [d for d in data if d[\"first_arg1_words\"] != [''] and d[\"first_arg2_words\"] != ['']\n",
    "       and d[\"second_arg1_words\"] != [''] and d[\"second_arg2_words\"] != ['']]\n",
    "data = [d for d in data if '' not in d[\"first_arg1_words\"] and '' not in d[\"first_arg2_words\"]\n",
    "       and '' not in d[\"second_arg1_words\"] and '' not in d[\"second_arg2_words\"]]\n",
    "\n",
    "def is_number_between_brackets(sent, ind):\n",
    "    lst = sent.split(\" \")\n",
    "    if not is_number(lst[ind]): return False\n",
    "    if ind == len(lst) - 1 or ind == 0: return False\n",
    "    return (lst[ind -1] == \"[\" and lst[ind + 1] == \"]\") or (lst[ind -1] == \"(\" and lst[ind + 1] == \")\")\n",
    "\n",
    "def is_number(s):\n",
    "    try:\n",
    "        d = int(s)\n",
    "        return True\n",
    "    except:\n",
    "        return False\n",
    "\n",
    "data = [d for d in data if not is_number_between_brackets(d[\"second\"], d[\"second_arg1\"][0]) and not\n",
    "       is_number_between_brackets(d[\"second\"], d[\"second_arg2\"][0]) and not\n",
    "        is_number_between_brackets(d[\"first\"], d[\"first_arg1\"][0]) and not\n",
    "        is_number_between_brackets(d[\"first\"], d[\"first_arg2\"][0])]\n",
    "\n",
    "np.random.seed(0)\n",
    "ids_subset = np.random.choice(list(ids), size = int(0.2 * len(ids)), replace = False)\n",
    "data_train = [d for d in data if d[\"query_id\"] not in ids_subset]\n",
    "data_dev = [d for d in data if d[\"query_id\"] in ids_subset]\n",
    "print(ids_subset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11594 55515\n"
     ]
    }
   ],
   "source": [
    "same = [d for d in data if d[\"query_first\"] == d[\"query_second\"]]\n",
    "not_same = [d for d in data if d[\"query_first\"] != d[\"query_second\"]]\n",
    "print(len(same), len(not_same))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'first': 'RESULTS : Acidosis limited the cellular consumption of glucose and ATP , causing tumor cells to <<ARG1:enter a metabolically dormant but energetically economic state>> , which promoted tumor cell survival during {{ARG2:glucose deficiency}} .',\n",
       " 'second': 'How might clonal hematopoiesis in aged populations increase their risk of death from COVID-19 infection ?',\n",
       " 'second_with_arguments': 'How might <<ARG1:clonal hematopoiesis>> in aged populations increase their risk of death from {{ARG2:COVID-19 infection}} ?',\n",
       " 'query_first': '<E|U+C>arg1:something, that improved $survival in <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION|CANCER]something',\n",
       " 'query_second': '<E|U+C>arg1:something can $[w=decrease|prevent|increase]decrease $death rates in <E|U+C>arg2:[e=DISEASE|CANCER|PATHOLOGICAL_FORMATION]something',\n",
       " 'first_arg1': (16, 23),\n",
       " 'first_arg2': (31, 32),\n",
       " 'second_arg1': (2, 3),\n",
       " 'second_arg2': (13, 14),\n",
       " 'first_arg1_words': ['glucose', 'deficiency'],\n",
       " 'first_arg2_words': ['glucose', 'deficiency'],\n",
       " 'second_arg1_words': ['clonal', 'hematopoiesis'],\n",
       " 'second_arg2_words': ['COVID-19', 'infection'],\n",
       " 'query_id': 25}"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d = data_train[330]\n",
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'first': '<<ARG1:Cerebral venous sinus thrombosis>> is a rare complication of {{ARG2:retrosigmoid resection of CPA tumor}} .',\n",
       " 'second': 'Other complications caused by cataracts are lens subluxation or luxation and glaucoma .',\n",
       " 'second_with_arguments': 'Other complications caused by {{ARG2:cataracts>> are <<ARG1:lens subluxation or luxation and glaucoma}} .',\n",
       " 'query_first': '<E|U+C>arg1:[e]stroke is a $complication of <E|U+C>arg2:flu :vaccination',\n",
       " 'query_second': 'among the $complications trigerred $by <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma is <E|U+C>arg1:[e]stroke',\n",
       " 'first_arg1': (0, 3),\n",
       " 'first_arg2': (9, 13),\n",
       " 'second_arg1': (6, 11),\n",
       " 'second_arg2': (4, 4),\n",
       " 'first_arg1_words': ['retrosigmoid', 'resection', 'of', 'CPA', 'tumor'],\n",
       " 'first_arg2_words': ['retrosigmoid', 'resection', 'of', 'CPA', 'tumor'],\n",
       " 'second_arg1_words': ['lens',\n",
       "  'subluxation',\n",
       "  'or',\n",
       "  'luxation',\n",
       "  'and',\n",
       "  'glaucoma'],\n",
       " 'second_arg2_words': ['cataracts'],\n",
       " 'query_id': 7}"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_same[20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"data.pickle\", \"wb\") as f:\n",
    "    pickle.dump(data, f)\n",
    "    \n",
    "with open(\"data_dev.pickle\", \"wb\") as f:\n",
    "    pickle.dump(data_dev, f)\n",
    "    \n",
    "with open(\"data_train.pickle\", \"wb\") as f:\n",
    "    pickle.dump(data_train, f)\n",
    "    \n",
    "import json\n",
    "\n",
    "def write(data, fname):\n",
    "    \n",
    "    with open(fname, \"w\") as f:\n",
    "        for d in data:\n",
    "            first, second = d[\"first\"], d[\"second_with_arguments\"]\n",
    "            first_arg1 = d[\"first_arg1\"]\n",
    "            first_arg2 = d[\"first_arg2\"]\n",
    "            second_arg1 = d[\"second_arg1\"]\n",
    "            second_arg2 = d[\"second_arg2\"]\n",
    "        \n",
    "            elems = [first, second, first_arg1, first_arg2, second_arg1, second_arg2]\n",
    "            keys = [\"first\", \"second\", \"first_arg1\", \"first_arg2\"]\n",
    "        \n",
    "            f.write(json.dumps(d) + \"\\n\")\n",
    "        \n",
    "            #f.write(d[\"first\"] + \"\\t\" + d[\"second\"] + \"\\t\" + d[\"query\"] + \"\\t\" + \"-\".join(d[\"first_arg1\"])+ \"\\n\")\n",
    "            \n",
    "    \n",
    "write(data, \"data.txt\")\n",
    "write(data_dev, \"data_dev.txt\")\n",
    "write(data_train, \"data_train.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(55459, 11650)"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_train), len(data_dev)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<<ARG1:Old World monkeys ( OWM , referring to Africa and Asia ) and hominoid primates>> {{ARG2:hominoid primates}} ( apes , humans ) , share a common ancestor . (0, 14) (13, 14)\n",
      "This indicates that the tremoviruses , <<ARG1:hepatovirus and lineage>> {{ARG2:lineage 1}} viruses have a common ancestor and evolved in their natural hosts , independently .\n",
      "---------------\n",
      "<E|U+C>arg1:something $and <E|U+C>arg2:something share $a $common $[w=origin|ancestry|ancestor]ancestry\n",
      "<E|U+C>arg1:something $and <E|U+C>arg2:something share $a $common $[w=origin|ancestry|ancestor]ancestry\n",
      "============================\n",
      "The zoonosis caused by <<ARG1:Dirofilaria immitis>> , a parasite of {{ARG2:dogs and other mammals}} , is transmitted by mosquitos and black flies to humans . (4, 5) (10, 13)\n",
      "N. gonorrhoeae infects columnar cells ; <<ARG1:it>> is therefore a parasite of {{ARG2:the cervix}} , not the vagina .\n",
      "---------------\n",
      "<E|U+C>arg1:something, $a <E|U+C>arg2:[e]bats $parasite\n",
      "<E|U+C>arg1:something $is $a <E|U+C>arg2:[e=ORGANISM|ORGAN|ORGANISM_SUBSTANCE]bats $parasite\n",
      "============================\n",
      "<<ARG1:ALF>> due to AIH with encephalopathy usually does not respond to any form of immunosuppression and needs {{ARG2:urgent liver transplant}} [ 27 ] . (0, 0) (17, 19)\n",
      "<<ARG1:Kidney transplant>> is the definitive treatment for {{ARG2:end stage renal disease}} ( ESRD ) .\n",
      "---------------\n",
      "<E|U+C>arg1:[e=DISEASE]disease may $[l=require|necessitate|need]require <E|U+C>arg2:[e=ORGAN]lung $transplant\n",
      "<E|U+C>arg1:[e=ORGAN]Liver $transplant is a $[l=treatment|measure|operation]treatment for <E|U+C>arg2:[e=DISEASE]flu\n",
      "============================\n",
      "{{ARG2:The reason>> for the different resistance levels might be the alternative mode of action of <<ARG1:glyphosate ( inhibitor of aromatic amino acid biosynthesis ) and isoxaflutole}} ( inhibitor of tocopherol-and plastoquinonebiosynthesis ) . (15, 25) (0, 1)\n",
      "<<ARG1:The cytotoxic effect of arsenic>> is at least partially attributed to the generation of {{ARG2:ROS}} .\n",
      "---------------\n",
      "<E|U+C>arg2:inhibition of kinases is the $[lemma=mode|mechanism]mode $of $action of <E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol\n",
      "the $[lemma=effect|influence|action]effect of <E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol is $attributed to <E|U+C>arg2:ligand activation\n",
      "============================\n",
      "Likewise , when <<ARG1:a wet cough>> persists after {{ARG2:4}} weeks of appropriate antibiotics , CHEST guidelines also suggest performing \" further investigations ( e.g. , flexible bronchoscopy with quantitative cultures and sensitivities with or without chest computed tomography ) \" ( 3 ) . (3, 5) (8, 8)\n",
      "The persistence of <<ARG1:influenza virus , rhinovirus , and norovirus>> was reported at {{ARG2:1 - 2}} days , 2 h −7 days , and 8 h −7 days , respectively [ 41 ] .\n",
      "---------------\n",
      "<E|U+C>arg1:[e=DISEASE|ORGANISM]COVID-19 $[l]persists for <E|U+C>arg2:10 $[lemma=day|week|month]days\n",
      "a study shows $persistence of <E|U+C>arg1:flu $virus for <E|U+C>arg2:10 $[lemma=day|week|month]days\n",
      "============================\n",
      "SARS is caused by a new type of <<ARG1:coronavirus>> , termed SARS-CoV , that may have evolved from {{ARG2:bat coronaviruses [ 10}} ] . (8, 8) (18, 21)\n",
      "Phylogenomic studies to determine evolutionary relationship of bats with other mammals remain unresolved with confl icting studies concluding <<ARG1:either bats>> diverged from {{ARG2:horses or from a sister}} a group within the clade of ungulates , cetaceans and carnivores ( Wynne and Tachedjian 2015 ) .\n",
      "---------------\n",
      "<E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]flu, that $[l=evolve|diverge]evolved $from <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]something\n",
      "<E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]flu $[l=evolve|diverge]evolved $from <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE|COVID-19]something\n",
      "============================\n",
      "The greater antibacterial effect of <<ARG1:N , O-CMC NPs>> was attributed to their relatively higher degree of {{ARG2:substitution of carboxymethyl groups}} on chitosan than unmodified chitosan , ( Sun et al. 2006 ) . (5, 8) (17, 20)\n",
      "<<ARG1:N -acetylcysteine>> ( NAC ) , the standard antidote for acetaminophen overdose , exerts its therapeutic effects by restoration of {{ARG2:depleted hepatic glutathione stores}} [ 38 ] ( Table 25 . 2 ) .\n",
      "---------------\n",
      "the $[lemma=effect|influence|action]effect of <E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol is $attributed to <E|U+C>arg2:ligand activation\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol $exerts its influnce $by <E|U+C>arg2:ligand activation\n",
      "============================\n",
      "Richter et al 19 19 It is currently difficult to determine whether patients with <<ARG1:IPF>> are more susceptible to {{ARG2:infection or colonization}} due to abnormal lung parenchyma , associated traction bronchiectasis , and immunosuppressive medications ( which historically have been commonly used for treatment of ILDs ) , or whether bacteria are involved as triggers of AE or in the pathogenesis of IPF . (14, 14) (19, 21)\n",
      "Background : {{ARG2:Cystic fibrosis-related diabetes>> ( CFRD ) is a known comorbidity in some patients with <<ARG1:cystic fibrosis}} .\n",
      "---------------\n",
      "people with <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma are $susceptible to <E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\n",
      "<E|U+C>arg2:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma :is a common $comorbidity in people with <E|U+C>arg1:[e=DISEASE|PATHOLOGICAL_FORMATION]asthma\n",
      "============================\n",
      "<<ARG1:Lipoteichoic acid>> ( LTA ) is a cellular toxic agent derived from {{ARG2:gram-positive bacteria}} . (0, 1) (12, 13)\n",
      "<<ARG1:Curcumin[(1E,6E)-1,7bis(4-hydroxy-3-methoxyphenyl)-1,6heptadiene-3,5-dione>> ] is a natural medicine mainly extracted from {{ARG2:plants of the Curcuma longa}} that has a long history to be used in humans in treating diseases without overt side effects .\n",
      "---------------\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]drug is a $[l=drug|medicine|agent|molecule]drug extracted $from <E|U+C>arg2:[l=plant|yeast|bacteria|soil|fungus|fungi|bark|leaf]plants\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]drug is a $[l=drug|medicine|agent|molecule]drug extracted $from <E|U+C>arg2:[l=plant|yeast|bacteria|soil|fungus|fungi|bark|leaf]plants\n",
      "============================\n",
      "<<ARG1:Porcine parvovirus infection>> induces apoptosis in PK-15 cells through activation of p53 and mitochondria-mediated pathway SARS-CoV 3a protein causes {{ARG2:p38 MAPK-mediated upregulation of p53}} , which plays an important role in 3a protein-mediated apoptosis ( Mizutani et al. , 2004 ) . (0, 2) (19, 23)\n",
      "<<ARG1:SARS-CoV-2 infection>> causes {{ARG2:dry cough , fever , shortness of breath , and acute respiratory distress syndrome}} , which may lead to death [ 2 ] .\n",
      "---------------\n",
      "<E|U+C>arg1:virus $infection $causes a <E|U+C>arg2:condition\n",
      "<E|U+C>arg1:virus $infection $causes a <E|U+C>arg2:condition\n",
      "============================\n",
      "<<ARG1:Lipoteichoic acid>> ( LTA ) is a cellular toxic agent derived from {{ARG2:gram-positive bacteria}} . (0, 1) (12, 13)\n",
      "<<ARG1:The Brevinin peptides>> are antimicrobial agents obtained from {{ARG2:frog skin secretions}} .\n",
      "---------------\n",
      "<E|U+C>arg1:something is a $[l=drug|medicine|agent]drug $[l=extract|distill|obtain|derive]extracted from <E|U+C>arg2:plant\n",
      "<E|U+C>arg1:something is a $[l=drug|medicine|agent]drug $[l=extract|distill|obtain|derive]extracted from <E|U+C>arg2:plant\n",
      "============================\n",
      "<<ARG1:SARS-CoV-2>> remains viable on {{ARG2:different types of surfaces}} , such as metal and plastic , for up to 72 hours , depending on the type of surface material , and can remain infectious in aerosols for at least 3 hours ( Van Doremalen et al. , 2020 ) . (0, 0) (4, 7)\n",
      "Experimental studies have shown that <<ARG1:SARS-CoV-2>> remains infectious in aerosols for hours [ 94 ] and on {{ARG2:various surfaces}} up to days , even though the virus is susceptible to standard disinfection procedures [ 94 , 95 ] .\n",
      "---------------\n",
      "<E|U+C>arg1:[e]COVID-19 $[l]remains $viable $on <E|U+C>arg2:surfaces\n",
      "<E|U+C>arg1:[e]COVID-19 $[l]remains $infectious $on <E|U+C>arg2:surfaces\n",
      "============================\n",
      "<<ARG1:The interferon-induced transmembrane ( IFITM ) protein family ( also known as the Fragilis family in mice )>> is a part of {{ARG2:the dispanin family}} [ 1 ] and is composed of double-transmembrane α-helices connected by a cytoplasmic ( CP ) loop and extracellular ( EC ) amino-and carboxyl-terminal polypeptide sequences ( Figure 1-A ) . (0, 17) (22, 24)\n",
      "<<ARG1:Varicella zoster virus>> ( VZV ) , known to cause chickenpox , is a part of {{ARG2:Herpesviridae family}} .\n",
      "---------------\n",
      "the <E|U+C>arg1:[e]MERS-CoV $is $a $part $of <E|U+C>arg2:something $family\n",
      "the <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]COVID-19 $is $a $part $of <E|U+C>arg2:something $family\n",
      "============================\n",
      "This third phase was attributed to a treatment-enhanced loss of HCV-infected cells . and found that {{ARG2:a triphasic HCV RNA decay pattern>> was more common in <<ARG1:HCV genotypes 4 and 1 ( 38 % ) versus genotypes 2 and 3}} ( 3 % ) . (26, 39) (16, 21)\n",
      "In a further study , the same group used <<ARG1:the optimized F gene>> to test a DNA vaccine in mice , showing induction of antibodies and protection against {{ARG2:HRSV}} ( Ternette et al. , 2007b ) .\n",
      "---------------\n",
      "<E|U+C>arg2:something is $more $common in <E|U+C>arg1:SCD $[l]genotypes\n",
      "<E|U+C>arg1:[e=DNA|GENE_OR_GENE_PRODUCT]SCD provide $protection against <E|U+C>arg2:something\n",
      "============================\n",
      "1 It is unknown whether <<ARG1:SARS-CoV-2>> transmitted directly from {{ARG2:bats}} to humans or through an intermediate host . (5, 5) (9, 9)\n",
      "If their ' fresh ' host had no defences , the viruses could cause a serious pandemic , as seen with HIV , SARS ( severe acute respiratory syndrome ) and <<ARG1:avian influenza virus>> that originated from chimpanzees , {{ARG2:the common palm civet and birds}} , respectively .\n",
      "---------------\n",
      "<E|U+C>arg1:something $transmitted $from <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]something\n",
      "The <E|U+C>arg1:flu $virus, which $originated in <E|U+C>arg2:[e=ORGANISM|ORGANISM_SUBSTANCE]bats\n",
      "============================\n",
      "It is wellestablished that <<ARG1:estradiol>> exerts important vasoactive effects by activation of {{ARG2:eNOS , as well as modulation of vasoactive substances}} released from endothelial cells or the direct vasodilation effect via relaxation of smooth muscle cells ( 92 ) . (4, 4) (12, 20)\n",
      "{{ARG2:This>> also is the mode of action of a number of <<ARG1:laxatives}} , including senna and castor oil , which contains the hydroxy fatty acid ricinoleic acid .\n",
      "---------------\n",
      "<E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol $exerts its influnce $by <E|U+C>arg2:ligand activation\n",
      "<E|U+C>arg2:inhibition of kinases is the $[lemma=mode|mechanism]mode $of $action of <E|U+C>arg1:[e=CHEMICAL|SIMPLE_CHEMICAL]paracetamol\n",
      "============================\n",
      "In contrast , <<ARG1:the AI H5N1 virus>> can survive at {{ARG2:25}} • C for 24 h , and at 40 • C for 15 min ( Chumpolbanchorn et al. , 2006 ) . (3, 6) (10, 10)\n",
      "In contrast , <<ARG1:the AI H5N1 virus>> can survive at 25 • C for {{ARG2:24}} h , and at 40 • C for 15 min ( Chumpolbanchorn et al. , 2006 ) .\n",
      "---------------\n",
      "<E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE|COVID-19]flu $can $survive $on surfaces for <E|U+C>arg2:10 days\n",
      "<E|U+C>arg1:flu $virus $can $survive $on surfaces for <E|U+C>arg2:10 days\n",
      "============================\n",
      "<<ARG1:Fasting blood glucose levels and HbA1c values>> in patients treated with antidiabetic drugs could be further reduced by {{ARG2:10 %}} upon coadministration of colesevelam ( Goldberg et al. 2008 ) . (0, 6) (18, 19)\n",
      "* Hypertension is a common presentation in COVID-19 due to the interactions of the virus with the angiotensin converting enzyme ( ACE2 ) : <<ARG1:a false normal blood pressure>> could be early signs of {{ARG2:deterioration}} in a previously hypertensive COVID-19 presentation a. # Sepsis with white cell count > 15x10 ∧ 9/L are relative contraindications for CNB , COVID-19 patients might present with lymphopenia and cardinal manifestations of sepsis may not be seen .\n",
      "---------------\n",
      "<E|U+C>arg1:glocuse $blood $levels are $[l=reduce|elevate]elevated in <E|U+C>arg2:something\n",
      "low $blood <E|U+C>arg1:glocuse is a $[l=sign|indication|symptom]sign of <E|U+C>arg2:something\n",
      "============================\n",
      "It was reported that the expression of ssc-miR-21 - 5p was upregulated in PK-15 cells after pseudorabies virus ( PRV ) infection , which inhibited <<ARG1:the replication of the PRV>> by directly targeting {{ARG2:chemokine ( C-X-C motif ) ligand 10}} ( CXCL10 ) , also named interferon-γ inducible protein-10 ( IP-10 ) [ 104 ] . (25, 29) (33, 39)\n",
      "<<ARG1:The SARS-CoV-2 infection>> induces {{ARG2:severe immune system ( IS ) dysfunctions}} , including spleen and lymph node atrophy , functional depletion of T lymphocytes , manifested by a decreased number of Th and Ts lymphocytes in secondary lymphoid organs , especially in critically ill patients .\n",
      "---------------\n",
      "<E|U+C>arg1:something $virus affects the <E|U+C>arg2:something $ligand\n",
      "<E|U+C>arg1:something $[l]infection affects the <E|U+C>arg2:[e]respiratory $system\n",
      "============================\n",
      "SARS-CoV was transmitted to {{ARG2:26 HCWs>> from <<ARG1:7 patients}} ; 21 HCWs were infected by 3 patients . (7, 8) (4, 5)\n",
      "Transboundary transmission of zoonotic agents from <<ARG1:Egypt>> to {{ARG2:Europe , Asia and America}} occurred via different pathways .\n",
      "---------------\n",
      "something, $[l]transmitted $to <E|U+C>arg2:someone $from <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBDIVISION|ORGANISM_SUBSTANCE]someone\n",
      "$transmission $of something $from <E|U+C>arg1:[e=ORGANISM|ORGANISM_SUBSTANCE]bats $to <E|U+C>arg2:someone\n",
      "============================\n"
     ]
    }
   ],
   "source": [
    "for i in range(20):\n",
    "    print(data[i][\"first\"], data[i][\"first_arg1\"], data[i][\"first_arg2\"])\n",
    "    print(data[i][\"second_with_arguments\"])\n",
    "    print(\"---------------\")\n",
    "    print(data[i][\"query_first\"])\n",
    "    print(data[i][\"query_second\"])\n",
    "    print(\"============================\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "67045\n"
     ]
    }
   ],
   "source": [
    "print(len(data))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "lengths = [len(d[\"first\"].split(\" \")) + len(d[\"second\"].split(\" \")) for d in data]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(67.92299198429576, 22.057857454680867, 237, 16)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.mean(lengths), np.std(lengths), np.max(lengths), np.min(lengths)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len([l for l in lengths if l > 250])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
